<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Generate hypothetical transcripts by adding/removing exons — Add-remove-exons • factR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Generate hypothetical transcripts by adding/removing exons — Add-remove-exons" />
<meta property="og:description" content="Adding or removing alternative segments to existing transcripts allow 
functional investigation of hypothetical RNAs. The functions below accepts
two vectors of the same length: x being a GRangesList containing exon coordinates
for each transcript to be modified, and y being a GRanges object containing
exon coordinates to add/remove to/from transcripts in x. The operations are done
in parallel, i.e. first exon in y will be added/remove from the first transcript
in x." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">factR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/factR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate hypothetical transcripts by adding/removing exons</h1>
    
    <div class="hidden name"><code>Add-remove-exons.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Adding or removing alternative segments to existing transcripts allow 
functional investigation of hypothetical RNAs. The functions below accepts
two vectors of the same length: x being a GRangesList containing exon coordinates
for each transcript to be modified, and y being a GRanges object containing
exon coordinates to add/remove to/from transcripts in x. The operations are done
in parallel, i.e. first exon in y will be added/remove from the first transcript
in x.</p>
    </div>

    <pre class="usage"><span class='fu'>addExonstoTx</span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>y</span>, drop.unmodified <span class='op'>=</span> <span class='cn'>FALSE</span>, allow.external.exons <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='fu'>removeExonsfromTx</span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>y</span>, drop.unmodified <span class='op'>=</span> <span class='cn'>FALSE</span>, ignore.strand <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>GRangesList containing exon coordinates of each transcript to be modified</p></td>
    </tr>
    <tr>
      <th>y</th>
      <td><p>GRanges object containing exon coordinates to be added/removed in parallel
to transcripts in x. Length of y and x have to be equal. In addition, x-y pairs
have to be on the same chromosome and strand.</p></td>
    </tr>
    <tr>
      <th>drop.unmodified</th>
      <td><p>Whether to drop hypothetical transcripts which were not successfully modified.
(Default; FALSE)</p></td>
    </tr>
    <tr>
      <th>allow.external.exons</th>
      <td><p>Whether to allow insertion of exons that fall outside the range of the transcript
(Default: FALSE). We recommend carefully inspecting the hypothetical transcripts 
when switching this argument to TRUE, as some exons may not originate from the
same gene.</p></td>
    </tr>
    <tr>
      <th>ignore.strand</th>
      <td><p>Whether to ignore strand when removing segments from transcript (Default: FALSE)</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>GRangesList containing modified transcripts in x. If drop.unmodified is FALSE,
output object is of same length as x and and contain metadata `modified` to
annotate whether insertion/removal is successful. All other metadata in x will 
not be retained</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='r-in'><span class='co'>## ---------------------------------------------------------------------</span></span>
<span class='r-in'><span class='co'>## EXAMPLE USING TOY DATASET</span></span>
<span class='r-in'><span class='co'>## ---------------------------------------------------------------------</span></span>
<span class='r-in'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>require</a></span><span class='op'>(</span><span class='va'><a href='https://bioconductor.org/packages/GenomicRanges'>GenomicRanges</a></span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: GenomicRanges</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: stats4</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: BiocGenerics</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: parallel</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Attaching package: ‘BiocGenerics’</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> The following objects are masked from ‘package:parallel’:</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     parLapplyLB, parRapply, parSapply, parSapplyLB</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> The following objects are masked from ‘package:stats’:</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     union, unique, unsplit, which.max, which.min</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: S4Vectors</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Attaching package: ‘S4Vectors’</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     expand.grid, I, unname</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: IRanges</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># create GRangesList of transcript exon coordinates</span></span>
<span class='r-in'><span class='va'>gr1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></span><span class='op'>(</span></span>
<span class='r-in'>seqnames <span class='op'>=</span> <span class='st'>"1"</span>, strand <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>rep</a></span><span class='op'>(</span><span class='st'>"+"</span>, <span class='fl'>2</span><span class='op'>)</span>,</span>
<span class='r-in'>ranges <span class='op'>=</span> <span class='fu'>IRanges</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html'>IRanges</a></span><span class='op'>(</span></span>
<span class='r-in'>start <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>501</span><span class='op'>)</span>,end <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>c</a></span><span class='op'>(</span><span class='fl'>100</span>, <span class='fl'>600</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>grl1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html'>GRangesList</a></span><span class='op'>(</span><span class='va'>gr1</span>, <span class='va'>gr1</span>, <span class='va'>gr1</span><span class='op'>)</span></span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>grl1</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>c</a></span><span class='op'>(</span><span class='st'>"tx1"</span>, <span class='st'>"tx2"</span>, <span class='st'>"tx3"</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># add casette exon, alternative 3'ss segment and retained intron</span></span>
<span class='r-in'><span class='va'>exons</span> <span class='op'>&lt;-</span> <span class='fu'>GenomicRanges</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></span><span class='op'>(</span></span>
<span class='r-in'>seqnames <span class='op'>=</span> <span class='st'>"1"</span>, strand <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>rep</a></span><span class='op'>(</span><span class='st'>"+"</span>, <span class='fl'>3</span><span class='op'>)</span>,</span>
<span class='r-in'>ranges <span class='op'>=</span> <span class='fu'>IRanges</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html'>IRanges</a></span><span class='op'>(</span>start <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>c</a></span><span class='op'>(</span><span class='fl'>201</span>, <span class='fl'>401</span>, <span class='fl'>101</span><span class='op'>)</span>,</span>
<span class='r-in'>                          end <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>c</a></span><span class='op'>(</span><span class='fl'>300</span>, <span class='fl'>500</span>, <span class='fl'>500</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span class='r-in'>                          </span>
<span class='r-in'><span class='va'>mod.grl1</span> <span class='op'>&lt;-</span> <span class='fu'>addExonstoTx</span><span class='op'>(</span><span class='va'>grl1</span>, <span class='va'>exons</span><span class='op'>)</span></span>
<span class='r-in'> </span>
<span class='r-in'><span class='co'># remove exons from hypothetical transcripts</span></span>
<span class='r-in'><span class='fu'>removeExonsfromTx</span><span class='op'>(</span><span class='va'>mod.grl1</span>, <span class='va'>exons</span><span class='op'>)</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRangesList object of length 3:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $tx1</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 2 ranges and 1 metadata column:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames    ranges strand |  modified</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;logical&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]        1     1-100      + |      TRUE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [2]        1   501-600      + |      TRUE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $tx2</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 2 ranges and 1 metadata column:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames    ranges strand |  modified</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;logical&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]        1     1-100      + |      TRUE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [2]        1   501-600      + |      TRUE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $tx3</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 2 ranges and 1 metadata column:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames    ranges strand |  modified</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;logical&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]        1     1-100      + |      TRUE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [2]        1   501-600      + |      TRUE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-in'></span>
<span class='r-in'></span>
<span class='r-in'></span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


