<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Reference-guided construction of CDS on GTF object — buildCDS • factR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Reference-guided construction of CDS on GTF object — buildCDS"><meta property="og:description" content="`buildCDS()` is designed to construct CDS information on transcripts from
query GTF object."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">factR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/quickstart.html">
    <span class="fas fa-rocket-launch"></span>
     
    Quick start
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fas fa-book-open"></span>
     
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/factR.html">1) factR workflow</a>
    </li>
    <li>
      <a href="../articles/analyzing_outputs.html">2) Analyzing factR output</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper"></span>
     
    News
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/fursham-h/factR" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/FurshamH" class="external-link">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Reference-guided construction of CDS on GTF object</h1>
    
    <div class="hidden name"><code>buildCDS.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>`buildCDS()` is designed to construct CDS information on transcripts from
query GTF object.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">buildCDS</span><span class="op">(</span><span class="va">query</span>, <span class="va">ref</span>, <span class="va">fasta</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>GRanges object containing query GTF data.</p></dd>


<dt>ref</dt>
<dd><p>GRanges object containing reference GTF data.</p></dd>


<dt>fasta</dt>
<dd><p>BSgenome or Biostrings object containing genomic sequence</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>GRanges object containing query exon entries and newly-constructed CDS
information</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The `buildCDS()`function will first search for known reference mRNAs in
`query` and annotate its CDS information. For the remaining transcripts, 
`buildCDS()` will search for a putative translation start site using a 
database of annotated ATG codons from `ref`. Transcripts containing an 
open-reading frame will be assigned the newly-determined CDS information.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Fursham Hamid</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Load genome and datasets</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Mmusculus.UCSC.mm10</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BSgenome</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘BiocGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:stats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, append,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     tapply, union, unique, unsplit, which.max, which.min</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: S4Vectors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: stats4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘S4Vectors’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     I, expand.grid, unname</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: IRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biostrings</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: XVector</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘Biostrings’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     strsplit</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: rtracklayer</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">matched_query_gtf</span>, <span class="va">ref_gtf</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Build CDS</span></span></span>
<span class="r-in"><span><span class="fu">buildCDS</span><span class="op">(</span><span class="va">matched_query_gtf</span>, <span class="va">ref_gtf</span>, <span class="va">Mmusculus</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Searching for reference mRNAs in query</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 reference mRNAs found and its CDS were assigned</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Building database of annotated ATG codons</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Selecting best ATG start codon for remaining transcripts and determining open-reading frame</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 new CDSs constructed</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Summary: Out of 4 transcripts in `matched_query_gtf`, </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>             4 transcript CDSs were built</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 105 ranges and 7 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames            ranges strand |       type transcript_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;factor&gt;   &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]    chr10 79854427-79864432      + | transcript   transcript1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]    chr10 79854427-79854721      + | exon         transcript1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]    chr10 79856504-79856534      + | exon         transcript1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]    chr10 79858752-79858824      + | exon         transcript1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]    chr10 79858952-79859271      + | exon         transcript1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...               ...    ... .        ...           ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [101]    chr10 79862014-79862047      + |        CDS   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [102]    chr10 79862449-79862541      + |        CDS   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [103]    chr10 79862653-79862869      + |        CDS   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [104]    chr10 79862978-79863055      + |        CDS   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [105]    chr10 79863145-79863274      + |        CDS   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       gene_id old_gene_id match_level   gene_name     phase</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   &lt;character&gt; &lt;character&gt;   &lt;numeric&gt; &lt;character&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1] ENSMUSG00000006498.14       GeneA           4       Ptbp1        NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2] ENSMUSG00000006498.14       GeneA           4       Ptbp1        NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3] ENSMUSG00000006498.14       GeneA           4       Ptbp1        NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4] ENSMUSG00000006498.14       GeneA           4       Ptbp1        NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5] ENSMUSG00000006498.14       GeneA           4       Ptbp1        NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...                   ...         ...         ...         ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [101] ENSMUSG00000006498.14        &lt;NA&gt;          NA       Ptbp1         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [102] ENSMUSG00000006498.14        &lt;NA&gt;          NA       Ptbp1         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [103] ENSMUSG00000006498.14        &lt;NA&gt;          NA       Ptbp1         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [104] ENSMUSG00000006498.14        &lt;NA&gt;          NA       Ptbp1         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [105] ENSMUSG00000006498.14        &lt;NA&gt;          NA       Ptbp1         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer></div>

  


  

  </body></html>

