<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Reference-guided construction of CDS on GTF object — buildCDS • factR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Reference-guided construction of CDS on GTF object — buildCDS" />
<meta property="og:description" content="`buildCDS()` is designed to construct CDS information on transcripts from
query GTF object." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">factR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/factR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Reference-guided construction of CDS on GTF object</h1>
    
    <div class="hidden name"><code>buildCDS.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>`buildCDS()` is designed to construct CDS information on transcripts from
query GTF object.</p>
    </div>

    <pre class="usage"><span class='fu'>buildCDS</span><span class='op'>(</span><span class='va'>query</span>, <span class='va'>ref</span>, <span class='va'>fasta</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>query</th>
      <td><p>GRanges object containing query GTF data.</p></td>
    </tr>
    <tr>
      <th>ref</th>
      <td><p>GRanges object containing reference GTF data.</p></td>
    </tr>
    <tr>
      <th>fasta</th>
      <td><p>BSgenome or Biostrings object containing genomic sequence</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>GRanges object containing query exon entries and newly-constructed CDS
information</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The `buildCDS()`function will first search for known reference mRNAs in
`query` and annotate its CDS information. For the remaining transcripts, 
`buildCDS()` will search for a putative translation start site using a 
database of annotated ATG codons from `ref`. Transcripts containing an 
open-reading frame will be assigned the newly-determined CDS information.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Fursham Hamid</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>BSgenome.Mmusculus.UCSC.mm10</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Loading required package: BSgenome</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocGenerics</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: parallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘BiocGenerics’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:parallel’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
#&gt; <span class='message'>    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
#&gt; <span class='message'>    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
#&gt; <span class='message'>    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
#&gt; <span class='message'>    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
#&gt; <span class='message'>    union, unique, unsplit, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: S4Vectors</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stats4</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘S4Vectors’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: IRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomeInfoDb</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Biostrings</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: XVector</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘Biostrings’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    strsplit</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: rtracklayer</span></div><div class='input'><span class='fu'>buildCDS</span><span class='op'>(</span><span class='va'>matched_query_gtf</span>, <span class='va'>ref_gtf</span>, <span class='va'>Mmusculus</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Searching for reference mRNAs in query</span></div><div class='output co'>#&gt; <span class='message'>2 reference mRNAs found and its CDS were assigned</span></div><div class='output co'>#&gt; <span class='message'>Building database of annotated ATG codons</span></div><div class='output co'>#&gt; <span class='message'>Selecting best ATG start codon for remaining transcripts and determining open-reading frame</span></div><div class='output co'>#&gt; <span class='message'>2 new CDSs constructed</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Summary: Out of 4 transcripts in `matched_query_gtf`, 4 transcript CDSs were built</span></div><div class='output co'>#&gt; GRanges object with 105 ranges and 7 metadata columns:
#&gt;         seqnames            ranges strand |       type transcript_id
#&gt;            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;factor&gt;   &lt;character&gt;
#&gt;     [1]    chr10 79854427-79864432      + | transcript   transcript1
#&gt;     [2]    chr10 79854427-79854721      + | exon         transcript1
#&gt;     [3]    chr10 79856504-79856534      + | exon         transcript1
#&gt;     [4]    chr10 79858752-79858824      + | exon         transcript1
#&gt;     [5]    chr10 79858952-79859271      + | exon         transcript1
#&gt;     ...      ...               ...    ... .        ...           ...
#&gt;   [101]    chr10 79862014-79862047      + |        CDS   transcript4
#&gt;   [102]    chr10 79862449-79862541      + |        CDS   transcript4
#&gt;   [103]    chr10 79862653-79862869      + |        CDS   transcript4
#&gt;   [104]    chr10 79862978-79863055      + |        CDS   transcript4
#&gt;   [105]    chr10 79863145-79863274      + |        CDS   transcript4
#&gt;                       gene_id old_gene_id match_level   gene_name     phase
#&gt;                   &lt;character&gt; &lt;character&gt;   &lt;numeric&gt; &lt;character&gt; &lt;numeric&gt;
#&gt;     [1] ENSMUSG00000006498.14       GeneA           4       Ptbp1        NA
#&gt;     [2] ENSMUSG00000006498.14       GeneA           4       Ptbp1        NA
#&gt;     [3] ENSMUSG00000006498.14       GeneA           4       Ptbp1        NA
#&gt;     [4] ENSMUSG00000006498.14       GeneA           4       Ptbp1        NA
#&gt;     [5] ENSMUSG00000006498.14       GeneA           4       Ptbp1        NA
#&gt;     ...                   ...         ...         ...         ...       ...
#&gt;   [101] ENSMUSG00000006498.14        &lt;NA&gt;          NA       Ptbp1         0
#&gt;   [102] ENSMUSG00000006498.14        &lt;NA&gt;          NA       Ptbp1         2
#&gt;   [103] ENSMUSG00000006498.14        &lt;NA&gt;          NA       Ptbp1         2
#&gt;   [104] ENSMUSG00000006498.14        &lt;NA&gt;          NA       Ptbp1         1
#&gt;   [105] ENSMUSG00000006498.14        &lt;NA&gt;          NA       Ptbp1         1
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


