<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Compare and classify alternative spliced segments between two or more transcripts — compareSplicedSegment • factR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Compare and classify alternative spliced segments between two or more transcripts — compareSplicedSegment" />
<meta property="og:description" content="Compare and classify alternative spliced segments between two or more transcripts" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">factR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/factR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Compare and classify alternative spliced segments between two or more transcripts</h1>
    
    <div class="hidden name"><code>compareSplicedSegment.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Compare and classify alternative spliced segments between two or more transcripts</p>
    </div>

    <pre class="usage"><span class='fu'>compareSplicedSegment</span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>Can be a GRangesList object containing exons for each transcripts (Intralist mode)</p>
<p>Can be a GRanges object containing exons for a transcript (Pair-wise mode). If so,
at least one GRanges object is to be provided in `...` for comparison</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>In pair-wise mode, argument is one or more GRanges object containing exons for a
particular transcript to compare with `x`</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>GRangesList object containing alternatively-spliced segments for each transcript
in comparison.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Fursham Hamid</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='r-in'><span class='co'>## ---------------------------------------------------------------------</span></span>
<span class='r-in'><span class='co'>## EXAMPLE USING TOY DATASET</span></span>
<span class='r-in'><span class='co'>## ---------------------------------------------------------------------</span></span>
<span class='r-in'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>require</a></span><span class='op'>(</span><span class='va'><a href='https://bioconductor.org/packages/GenomicRanges'>GenomicRanges</a></span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Create toy GRanges GTF object</span></span>
<span class='r-in'><span class='va'>gr1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></span><span class='op'>(</span><span class='st'>"chr1"</span>, <span class='fu'><a href='https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html'>IRanges</a></span><span class='op'>(</span>start <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>101</span><span class='op'>)</span>, width <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>c</a></span><span class='op'>(</span><span class='fl'>20</span>, <span class='fl'>20</span><span class='op'>)</span><span class='op'>)</span>, <span class='st'>"+"</span>, transcript_id <span class='op'>=</span> <span class='st'>"transcript1"</span>, gene_id <span class='op'>=</span> <span class='st'>"geneA"</span>, gene_name <span class='op'>=</span> <span class='st'>"gene001"</span><span class='op'>)</span></span>
<span class='r-in'><span class='va'>gr2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></span><span class='op'>(</span><span class='st'>"chr1"</span>, <span class='fu'><a href='https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html'>IRanges</a></span><span class='op'>(</span>start <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>51</span>, <span class='fl'>101</span><span class='op'>)</span>, width <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>c</a></span><span class='op'>(</span><span class='fl'>20</span>, <span class='fl'>10</span>, <span class='fl'>20</span><span class='op'>)</span><span class='op'>)</span>, <span class='st'>"+"</span>, transcript_id <span class='op'>=</span> <span class='st'>"transcript2"</span>, gene_id <span class='op'>=</span> <span class='st'>"geneA"</span>, gene_name <span class='op'>=</span> <span class='st'>"gene001"</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Pairwise comparison between GRanges object</span></span>
<span class='r-in'><span class='fu'>compareSplicedSegment</span><span class='op'>(</span><span class='va'>gr1</span>, <span class='va'>gr2</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRangesList object of length 1:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $transcript2</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 1 range and 5 metadata columns:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames    ranges strand |     gene_id   gene_name transcript_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;   &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]     chr1     51-60      + |       geneA       geneA   transcript2</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>              type      AStype</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       &lt;character&gt; &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]          AS          CE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Multiple comparisons can be done by providing more GRanges input</span></span>
<span class='r-in'><span class='va'>gr3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></span><span class='op'>(</span><span class='st'>"chr1"</span>, <span class='fu'><a href='https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html'>IRanges</a></span><span class='op'>(</span>start <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>91</span><span class='op'>)</span>, width <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>c</a></span><span class='op'>(</span><span class='fl'>20</span>, <span class='fl'>30</span><span class='op'>)</span><span class='op'>)</span>, <span class='st'>"+"</span>, transcript_id <span class='op'>=</span> <span class='st'>"transcript3"</span>, gene_id <span class='op'>=</span> <span class='st'>"geneA"</span>, gene_name <span class='op'>=</span> <span class='st'>"gene001"</span><span class='op'>)</span></span>
<span class='r-in'><span class='fu'>compareSplicedSegment</span><span class='op'>(</span><span class='va'>gr1</span>, <span class='va'>gr2</span>, <span class='va'>gr3</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRangesList object of length 2:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $transcript2</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 1 range and 5 metadata columns:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames    ranges strand |     gene_id   gene_name transcript_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;   &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]     chr1     51-60      + |       geneA       geneA   transcript2</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>              type      AStype</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       &lt;character&gt; &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]          AS          CE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $transcript3</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 1 range and 5 metadata columns:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames    ranges strand |     gene_id   gene_name transcript_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;   &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]     chr1    91-100      + |       geneA       geneA   transcript3</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>              type      AStype</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       &lt;character&gt; &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]          AS          SA</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## GRangesList containing exons per transcript can be given as input</span></span>
<span class='r-in'><span class='va'>grl</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html'>GRangesList</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>gr1</span>, <span class='va'>gr2</span>, <span class='va'>gr3</span><span class='op'>)</span><span class='op'>)</span></span>
<span class='r-in'><span class='fu'>compareSplicedSegment</span><span class='op'>(</span><span class='va'>grl</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRangesList object of length 2:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $transcript2</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 1 range and 5 metadata columns:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames    ranges strand |     gene_id   gene_name transcript_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;   &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]     chr1     51-60      + |       geneA       geneA   transcript2</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>              type      AStype</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       &lt;character&gt; &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]          AS          CE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $transcript3</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 1 range and 5 metadata columns:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames    ranges strand |     gene_id   gene_name transcript_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;   &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]     chr1    91-100      + |       geneA       geneA   transcript3</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>              type      AStype</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       &lt;character&gt; &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]          AS          SA</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-in'></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## ---------------------------------------------------------------------</span></span>
<span class='r-in'><span class='co'>## EXAMPLE USING SAMPLE DATASET</span></span>
<span class='r-in'><span class='co'>## ---------------------------------------------------------------------</span></span>
<span class='r-in'><span class='co'>## Using GRangesList as input</span></span>
<span class='r-in'><span class='fu'>compareSplicedSegment</span><span class='op'>(</span><span class='va'>query_exons</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRangesList object of length 4:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $transcript1</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 3 ranges and 5 metadata columns:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames            ranges strand |               gene_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |           &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]    chr10 79858952-79859071      + | ENSMUSG00000006498.14</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [2]    chr10 79860117-79860194      + | ENSMUSG00000006498.14</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [3]    chr10 79862014-79862047      + | ENSMUSG00000006498.14</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                   gene_name transcript_id        type      AStype</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                 &lt;character&gt;   &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1] ENSMUSG00000006498.14   transcript1          AS          SA</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [2] ENSMUSG00000006498.14   transcript1          AS          CE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [3] ENSMUSG00000006498.14   transcript1          AS          CE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $transcript2</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 2 ranges and 5 metadata columns:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames            ranges strand |               gene_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |           &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]    chr10 79858952-79859071      + | ENSMUSG00000006498.14</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [2]    chr10 79862014-79862047      + | ENSMUSG00000006498.14</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                   gene_name transcript_id        type      AStype</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                 &lt;character&gt;   &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1] ENSMUSG00000006498.14   transcript2          AS          SA</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [2] ENSMUSG00000006498.14   transcript2          AS          CE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $transcript3</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 1 range and 5 metadata columns:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames            ranges strand |               gene_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |           &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]    chr10 79858952-79859071      + | ENSMUSG00000006498.14</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                   gene_name transcript_id        type      AStype</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                 &lt;character&gt;   &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1] ENSMUSG00000006498.14   transcript3          AS          SA</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $transcript4</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 1 range and 5 metadata columns:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames            ranges strand |               gene_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |           &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]    chr10 79862014-79862047      + | ENSMUSG00000006498.14</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                   gene_name transcript_id        type      AStype</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                 &lt;character&gt;   &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1] ENSMUSG00000006498.14   transcript4          AS          CE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Compare AS between individual GRanges object</span></span>
<span class='r-in'><span class='fu'>compareSplicedSegment</span><span class='op'>(</span><span class='va'>query_exons</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span>, <span class='va'>query_exons</span><span class='op'>[[</span><span class='fl'>3</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRangesList object of length 1:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> $transcript1</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 2 ranges and 5 metadata columns:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames            ranges strand |               gene_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |           &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]    chr10 79860117-79860194      + | ENSMUSG00000006498.14</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [2]    chr10 79862014-79862047      + | ENSMUSG00000006498.14</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                   gene_name transcript_id        type      AStype</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                 &lt;character&gt;   &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1] ENSMUSG00000006498.14   transcript1          AS          CE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [2] ENSMUSG00000006498.14   transcript1          AS          CE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


