<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Find NMD-causing exons from GTF objects — identifyNMDexons • factR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Find NMD-causing exons from GTF objects — identifyNMDexons"><meta property="og:description" content="This function will identify alternative exons that triggers NMD when 
spliced or skipped."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">factR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/factR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Find NMD-causing exons from GTF objects</h1>
    
    <div class="hidden name"><code>identifyNMDexons.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function will identify alternative exons that triggers NMD when 
spliced or skipped.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">identifyNMDexons</span><span class="op">(</span><span class="va">x</span>, <span class="va">fasta</span>, NMD.result <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>x</dt>
<dd><p>GRanges object containing query GTF data with CDS information.</p></dd>
<dt>fasta</dt>
<dd><p>BSgenome or Biostrings object containing genomic sequence</p></dd>
<dt>NMD.result</dt>
<dd><p>Output dataframe from predictNMD() [Optional]. If not provided, or if 
data-frame do not contain necessary variables, identifyNMDexons will run
predictNMD() first.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>GRanges object containing coordinates of identified NMD-causing exons, 
including metadata on its gene of origin, type of AS exon, type of NMD exon
and whether it is spliced within CDS</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Alternative splicing of regulated exons may introduce premature termination 
codon which are cleared by nonsense-mediated decay. There are two major class 
of such exons: Poison and ORF-maintaining exons. The former triggers NMD when
spliced in, hence its name, while the latter triggers NMD when removed from a 
transcript.</p>
<p>To identify and classify NMD-causing exons, identifyNMDexons first find
alternative segments between NMD-sensitive transcripts and pre-determined 
reference transcript of the same gene. The longest, NMD-insensitive coding 
transcript of each gene is chosen as the best reference. For each alternative segment, 
a modified version of reference RNA is built by inserting exons spliced in NMD transcripts, or
by removing exons skipped from NMD transcripts. These modified transcripts are
then tested for NMD features and if positive, the alternative exon will be
annotated as NMD-causing.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co">## ---------------------------------------------------------------------</span></span>
<span class="r-in"><span class="co">## EXAMPLE USING SAMPLE DATASET</span></span>
<span class="r-in"><span class="co">## ---------------------------------------------------------------------</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Mmusculus.UCSC.mm10</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># build CDS information first</span></span>
<span class="r-in"><span class="va">out.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="buildCDS.html">buildCDS</a></span><span class="op">(</span><span class="va">matched_query_gtf</span>, <span class="va">ref_gtf</span>, <span class="va">Mmusculus</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Searching for reference mRNAs in query</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 reference mRNAs found and its CDS were assigned</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Building database of annotated ATG codons</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Selecting best ATG start codon for remaining transcripts and determining open-reading frame</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 new CDSs constructed</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Summary: Out of 4 transcripts in `matched_query_gtf`, 4 transcript CDSs were built</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># proceed with identifying NMD exons</span></span>
<span class="r-in"><span class="va">NMD.exons_1</span> <span class="op">&lt;-</span> <span class="fu">identifyNMDexons</span><span class="op">(</span><span class="va">out.gtf</span>, <span class="va">Mmusculus</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> `NMD.result` not provided, running predictNMD()</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Selecting best reference mRNAs</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in dplyr::slice_max(., order_by = ".", n = 1):</span> Problem while computing indices.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> The error occurred in group 1: gene_id = "ENSMUSG00000006498.14".</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Caused by error:</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> `order_by` must have size 83, not size 1.</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># if output of predictNMD() has been generated, this can be provided to</span></span>
<span class="r-in"><span class="co"># NMD.result argument</span></span>
<span class="r-in"><span class="va">NMD.output</span> <span class="op">&lt;-</span> <span class="fu"><a href="predictNMD.html">predictNMD</a></span><span class="op">(</span><span class="va">out.gtf</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Predicting NMD sensitivities for 4 mRNAs</span>
<span class="r-in"><span class="va">NMD.exons_2</span> <span class="op">&lt;-</span> <span class="fu">identifyNMDexons</span><span class="op">(</span><span class="va">out.gtf</span>, <span class="va">Mmusculus</span>, NMD.result <span class="op">=</span> <span class="va">NMD.output</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Selecting best reference mRNAs</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in dplyr::slice_max(., order_by = ".", n = 1):</span> Problem while computing indices.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> The error occurred in group 1: gene_id = "ENSMUSG00000006498.14".</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Caused by error:</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> `order_by` must have size 83, not size 1.</span>
<span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="va">NMD.exons_1</span>, <span class="va">NMD.exons_2</span><span class="op">)</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in identical(NMD.exons_1, NMD.exons_2):</span> object 'NMD.exons_1' not found</span>
<span class="r-in"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## ---------------------------------------------------------------------</span></span>
<span class="r-in"><span class="co">## EXAMPLE USING TRANSCRIPT ANNOTATION</span></span>
<span class="r-in"><span class="co">## ---------------------------------------------------------------------</span></span>
<span class="r-in"><span class="co"># \donttest{</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocFileCache</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: dbplyr</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘AnnotationHub’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:rtracklayer’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     hubUrl</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Retrieve GRCm38 trancript annotation</span></span>
<span class="r-in"><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class="r-in"><span class="va">GRCm38_gtf</span> <span class="op">&lt;-</span> <span class="va">ah</span><span class="op">[[</span><span class="st">"AH60127"</span><span class="op">]</span><span class="op">]</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> downloading 1 resources</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> retrieving 1 resource</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing File into R ..</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> '?repositories' for details</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> replacement repositories:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     CRAN: https://cloud.r-project.org</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> '?repositories' for details</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> replacement repositories:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     CRAN: https://cloud.r-project.org</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Identify NMD-causing exons in entire transcriptome</span></span>
<span class="r-in"><span class="va">NMD.exons.all</span> <span class="op">&lt;-</span> <span class="fu">identifyNMDexons</span><span class="op">(</span><span class="va">GRCm38_gtf</span>, <span class="va">Mmusculus</span><span class="op">)</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in .identifynmdchecks(x, fasta, NMD.result, argnames):</span> `GRCm38_gtf` and `Mmusculus` has unmatched seqlevel styles. </span>
<span class="r-err co"><span class="r-pr">#&gt;</span> Try running: GRCm38_gtf &lt;- matchChromosomes(GRCm38_gtf, Mmusculus)</span>
<span class="r-in"><span class="co"># }</span></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

