<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Match gene metadata from query GTF to a reference GTF — matchGeneInfo • factR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Match gene metadata from query GTF to a reference GTF — matchGeneInfo"><meta property="og:description" content="`matchGeneInfo()` matches and corrects Gene IDs from a 
query GTF object to a reference GTF"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">factR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/quickstart.html">
    <span class="fas fa-rocket-launch"></span>
     
    Quick start
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fas fa-book-open"></span>
     
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/factR.html">1) factR workflow</a>
    </li>
    <li>
      <a href="../articles/analyzing_outputs.html">2) Analyzing factR output</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper"></span>
     
    News
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/fursham-h/factR" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/FurshamH" class="external-link">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Match gene metadata from query GTF to a reference GTF</h1>
    
    <div class="hidden name"><code>matchGeneInfo.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>`matchGeneInfo()` matches and corrects Gene IDs from a 
query GTF object to a reference GTF</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">matchGeneInfo</span><span class="op">(</span><span class="va">query</span>, <span class="va">ref</span>, primary_gene_id <span class="op">=</span> <span class="cn">NULL</span>, secondary_gene_id <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>Query GTF imported as GRanges object</p></dd>


<dt>ref</dt>
<dd><p>Reference GTF as GRanges object</p></dd>


<dt>primary_gene_id</dt>
<dd><p>Character name of the primary gene id metadata in query GTF.
Input to this argument is typically 'gene_id'</p></dd>


<dt>secondary_gene_id</dt>
<dd><p>Character name of the secondary gene id in query file.
Example of input to this argument is 'ref_gene_id'</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Gene_id-matched query GRanges</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The default approach to this correction relies on finding overlaps 
between transcripts in query with transcripts in reference. Using this 
method alone could result in false positive matches 
(19 percent false positives).
To improve this, users have the option to invoke two additional layers 
of matching.  
(1) Matching by ENSEMBL Gene_IDs. If both query and reference transcript 
annotations containg Ensembl-style Gene IDs, this program will try to 
match both IDs in a less stringent manner. This correction can be invoked
by providing the 'primary_gene_id' argument</p>
<p>(2) Matching by secondary Gene_IDs. Depending on the transcript assembly 
program, GTF/GFF3 annotations may contain additional comments on the
transcript information. This may include a distinct secondary Gene ID 
annotation that potentially matches with the reference. To invoke this 
correction, provide 'primary_gene_id' and 'secondary_gene_id' arguments. 
To determine if your transcript assembly contain possible secondary 
Gene IDs, import query GTF file using `importGTF()` and check its metadata
columns</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Fursham Hamid</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## ---------------------------------------------------------------------</span></span></span>
<span class="r-in"><span><span class="co">## EXAMPLE USING SAMPLE DATASET</span></span></span>
<span class="r-in"><span><span class="co">## ---------------------------------------------------------------------</span></span></span>
<span class="r-in"><span><span class="co"># Load datasets</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">chrom_matched_query_gtf</span>, <span class="va">ref_gtf</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Run matching function</span></span></span>
<span class="r-in"><span><span class="fu">matchGeneInfo</span><span class="op">(</span><span class="va">chrom_matched_query_gtf</span>, <span class="va">ref_gtf</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Number of mismatched gene_ids found: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ---&gt; Attempting to match gene_ids by finding overlapping coordinates...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ---&gt; 1 gene_id matched</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total gene_ids corrected: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Remaining number of mismatched gene_ids: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 56 ranges and 6 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames            ranges strand |       type transcript_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;factor&gt;   &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]    chr10 79854427-79864432      + | transcript   transcript1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]    chr10 79854427-79854721      + | exon         transcript1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]    chr10 79856504-79856534      + | exon         transcript1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]    chr10 79858752-79858824      + | exon         transcript1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]    chr10 79858952-79859271      + | exon         transcript1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...               ...    ... .        ...           ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [52]    chr10 79862014-79862047      + |       exon   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [53]    chr10 79862449-79862541      + |       exon   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [54]    chr10 79862653-79862869      + |       exon   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [55]    chr10 79862978-79863055      + |       exon   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [56]    chr10 79863145-79864359      + |       exon   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      gene_id old_gene_id match_level   gene_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;character&gt; &lt;character&gt;   &lt;numeric&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...                   ...         ...         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [52] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [53] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [54] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [55] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [56] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer></div>

  


  

  </body></html>

