<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Predict NMD sensitivity on mRNA transcripts — predictNMD • factR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Predict NMD sensitivity on mRNA transcripts — predictNMD" />
<meta property="og:description" content="Predict NMD sensitivity on mRNA transcripts" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">factR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/factR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Predict NMD sensitivity on mRNA transcripts</h1>
    
    <div class="hidden name"><code>predictNMD.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Predict NMD sensitivity on mRNA transcripts</p>
    </div>

    <pre class="usage"><span class='fu'>predictNMD</span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>...</span>, cds <span class='op'>=</span> <span class='cn'>NULL</span>, NMD_threshold <span class='op'>=</span> <span class='fl'>50</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>Can be a GRanges object containing exon and CDS transcript features in GTF
format.</p>
<p>Can be a GRangesList object containing exon features for a list of transcripts.
If so, `cds` argument have to be provided.</p>
<p>Can be a GRanges object containing exon features for a transcript. If so, `cds`
argument have to be provided.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Logical conditions to pass to dplyr::filter to subset transcripts for analysis.
Variables are metadata information found in `x` and multiple conditions can be
provided delimited by comma. Example: transcript_id == "transcript1"</p></td>
    </tr>
    <tr>
      <th>cds</th>
      <td><p>If `x` is a GRangesList object, `cds` has to be a GRangesList containing CDS features
for the list of transcripts in `x`. List names in `x` and `cds` have to match.</p>
<p>If `x` is a GRanges object, `cds` has to be a GRanges containing CDS features
for the transcript in `x`.</p></td>
    </tr>
    <tr>
      <th>NMD_threshold</th>
      <td><p>Minimum distance of stop_codon to last exon junction (EJ) which triggers NMD.
Default = 50bp</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Dataframe with prediction of NMD sensitivity and NMD features:</p>
<p>is_NMD: logical value in prediciting transcript sensitivity to NMD</p>
<p>stop_to_lastEJ: Integer value of the number of bases between the first
base of the stop_codon to the last base of EJ. A positive value indicates that
the last EJ is downstream of the stop_codon.</p>
<p>num_of_down_EJs: Number of EJs downstream of the stop_codon.</p>
<p>`3_UTR_length`: Length of 3' UTR</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Fursham Hamid</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>## ---------------------------------------------------------------------</span>
<span class='co'>## EXAMPLE USING SAMPLE DATASET</span>
<span class='co'>## ---------------------------------------------------------------------</span>
<span class='co'>## Using GTF GRanges as input</span>
<span class='fu'>predictNMD</span><span class='op'>(</span><span class='va'>new_query_gtf</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Predicting NMD sensitivities for 4 mRNAs</span></div><div class='output co'>#&gt; <span style='color: #555555;'># A tibble: 4 x 5</span><span>
#&gt;   transcript  stop_to_lastEJ num_of_downEJs `3'UTR_length` is_NMD
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>                </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span> 
#&gt; </span><span style='color: #555555;'>1</span><span> transcript1           -</span><span style='color: #BB0000;'>130</span><span>              0           </span><span style='text-decoration: underline;'>1</span><span>158 FALSE 
#&gt; </span><span style='color: #555555;'>2</span><span> transcript2           -</span><span style='color: #BB0000;'>130</span><span>              0           </span><span style='text-decoration: underline;'>1</span><span>497 FALSE 
#&gt; </span><span style='color: #555555;'>3</span><span> transcript3            364              3            644 TRUE  
#&gt; </span><span style='color: #555555;'>4</span><span> transcript4           -</span><span style='color: #BB0000;'>130</span><span>              0           </span><span style='text-decoration: underline;'>1</span><span>085 FALSE </div><div class='input'>
<span class='co'>### Transcripts for analysis can be subsetted using logical conditions</span>
<span class='fu'>predictNMD</span><span class='op'>(</span><span class='va'>new_query_gtf</span>, <span class='va'>transcript_id</span> <span class='op'>==</span> <span class='st'>"transcript1"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Predicting NMD sensitivities for 1 mRNAs</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 1 x 5</span><span>
#&gt;   transcript stop_to_lastEJ num_of_downEJs `3'UTR_length` is_NMD
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>               </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span> 
#&gt; </span><span style='color: #555555;'>1</span><span> transcript           -</span><span style='color: #BB0000;'>130</span><span>              0           </span><span style='text-decoration: underline;'>1</span><span>158 FALSE </div><div class='input'><span class='fu'>predictNMD</span><span class='op'>(</span><span class='va'>new_query_gtf</span>, <span class='va'>transcript_id</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"transcript1"</span>, <span class='st'>"transcript3"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Predicting NMD sensitivities for 2 mRNAs</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 2 x 5</span><span>
#&gt;   transcript  stop_to_lastEJ num_of_downEJs `3'UTR_length` is_NMD
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>                </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span> 
#&gt; </span><span style='color: #555555;'>1</span><span> transcript1           -</span><span style='color: #BB0000;'>130</span><span>              0           </span><span style='text-decoration: underline;'>1</span><span>158 FALSE 
#&gt; </span><span style='color: #555555;'>2</span><span> transcript3            364              3            644 TRUE  </div><div class='input'>

<span class='co'>## Using exon and CDS GRangesLists as input</span>
<span class='fu'>predictNMD</span><span class='op'>(</span><span class='va'>query_exons</span>, cds <span class='op'>=</span> <span class='va'>query_cds</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Predicting NMD sensitivities for 4 mRNAs</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 4 x 5</span><span>
#&gt;   transcript  stop_to_lastEJ num_of_downEJs `3'UTR_length` is_NMD
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>                </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span> 
#&gt; </span><span style='color: #555555;'>1</span><span> transcript1           -</span><span style='color: #BB0000;'>130</span><span>              0           </span><span style='text-decoration: underline;'>1</span><span>158 FALSE 
#&gt; </span><span style='color: #555555;'>2</span><span> transcript2           -</span><span style='color: #BB0000;'>130</span><span>              0           </span><span style='text-decoration: underline;'>1</span><span>497 FALSE 
#&gt; </span><span style='color: #555555;'>3</span><span> transcript3            364              3            644 TRUE  
#&gt; </span><span style='color: #555555;'>4</span><span> transcript4           -</span><span style='color: #BB0000;'>130</span><span>              0           </span><span style='text-decoration: underline;'>1</span><span>085 FALSE </div><div class='input'><span class='fu'>predictNMD</span><span class='op'>(</span><span class='va'>query_exons</span>, cds <span class='op'>=</span> <span class='va'>query_cds</span>, <span class='va'>transcript_id</span> <span class='op'>==</span> <span class='st'>"transcript3"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Predicting NMD sensitivities for 1 mRNAs</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 1 x 5</span><span>
#&gt;   transcript stop_to_lastEJ num_of_downEJs `3'UTR_length` is_NMD
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>               </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span> 
#&gt; </span><span style='color: #555555;'>1</span><span> transcript            364              3            644 TRUE  </div><div class='input'>

<span class='co'>## Using exon and CDS GRanges as input</span>
<span class='fu'>predictNMD</span><span class='op'>(</span><span class='va'>query_exons</span><span class='op'>[[</span><span class='fl'>3</span><span class='op'>]</span><span class='op'>]</span>, cds <span class='op'>=</span> <span class='va'>query_cds</span><span class='op'>[[</span><span class='fl'>3</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Predicting NMD sensitivities for 1 mRNAs</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 1 x 5</span><span>
#&gt;   transcript stop_to_lastEJ num_of_downEJs `3'UTR_length` is_NMD
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>               </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span> 
#&gt; </span><span style='color: #555555;'>1</span><span> transcript            364              3            644 TRUE  </div><div class='input'>


<span class='co'>## ---------------------------------------------------------------------</span>
<span class='co'>## EXAMPLE USING TRANSCRIPT ANNOTATION</span>
<span class='co'>## ---------------------------------------------------------------------</span>
<span class='co'># \donttest{</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>AnnotationHub</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocFileCache</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: dbplyr</span></div><div class='input'>
<span class='co'>## Retrieve GRCm38 trancript annotation</span>
<span class='va'>ah</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html'>AnnotationHub</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>using temporary cache /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpyOQ32I/BiocFileCache</span></div><div class='output co'>#&gt; <span class='message'>snapshotDate(): 2020-10-27</span></div><div class='input'><span class='va'>GRCm38_gtf</span> <span class='op'>&lt;-</span> <span class='va'>ah</span><span class='op'>[[</span><span class='st'>"AH60127"</span><span class='op'>]</span><span class='op'>]</span>
</div><div class='output co'>#&gt; <span class='message'>downloading 1 resources</span></div><div class='output co'>#&gt; <span class='message'>retrieving 1 resource</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; <span class='message'>Importing File into R ..</span></div><div class='input'>
<span class='co'>## Run tool on specific gene family</span>
<span class='fu'>predictNMD</span><span class='op'>(</span><span class='va'>GRCm38_gtf</span>, <span class='va'>gene_name</span> <span class='op'>==</span> <span class='st'>"Ptbp1"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Predicting NMD sensitivities for 9 mRNAs</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 9 x 5</span><span>
#&gt;   transcript         stop_to_lastEJ num_of_downEJs `3'UTR_length` is_NMD
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>                       </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span> 
#&gt; </span><span style='color: #555555;'>1</span><span> ENSMUST00000057343            364              3            644 TRUE  
#&gt; </span><span style='color: #555555;'>2</span><span> ENSMUST00000095457           -</span><span style='color: #BB0000;'>130</span><span>              0           </span><span style='text-decoration: underline;'>1</span><span>085 FALSE 
#&gt; </span><span style='color: #555555;'>3</span><span> ENSMUST00000165704           -</span><span style='color: #BB0000;'>130</span><span>              0           </span><span style='text-decoration: underline;'>1</span><span>497 FALSE 
#&gt; </span><span style='color: #555555;'>4</span><span> ENSMUST00000165724            286              2            362 TRUE  
#&gt; </span><span style='color: #555555;'>5</span><span> ENSMUST00000168683           -</span><span style='color: #BB0000;'>145</span><span>              0              0 FALSE 
#&gt; </span><span style='color: #555555;'>6</span><span> ENSMUST00000169091            -</span><span style='color: #BB0000;'>36</span><span>              0              0 FALSE 
#&gt; </span><span style='color: #555555;'>7</span><span> ENSMUST00000169483            139              1            152 TRUE  
#&gt; </span><span style='color: #555555;'>8</span><span> ENSMUST00000171599            -</span><span style='color: #BB0000;'>63</span><span>              0              0 FALSE 
#&gt; </span><span style='color: #555555;'>9</span><span> ENSMUST00000172282           -</span><span style='color: #BB0000;'>130</span><span>              0           </span><span style='text-decoration: underline;'>1</span><span>158 FALSE </div><div class='input'><span class='co'># }</span>

</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


