<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Shortlist GTF GRanges object for new transcripts — subsetNewTranscripts • factR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Shortlist GTF GRanges object for new transcripts — subsetNewTranscripts"><meta property="og:description" content="`subsetNewTranscripts()` will retain transcripts in `query` that are distinct
from those in `ref`"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">factR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/factR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Shortlist GTF GRanges object for new transcripts</h1>
    
    <div class="hidden name"><code>subsetNewTranscripts.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>`subsetNewTranscripts()` will retain transcripts in `query` that are distinct
from those in `ref`</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">subsetNewTranscripts</span><span class="op">(</span><span class="va">query</span>, <span class="va">ref</span>, refine.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span>, <span class="st">"intron"</span>, <span class="st">"cds"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>GRanges object containing query GTF data.</p></dd>
<dt>ref</dt>
<dd><p>GRanges object containing reference GTF data.</p></dd>
<dt>refine.by</dt>
<dd><p>Whether to refine the selection process by removing query transcripts with
similar introns or CDS structure to reference. Default input is "none", and 
can be changed to "intron" or "cds" respectively.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Filtered GRanges GTF object</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>`subsetNewTranscripts()` will compare query and reference GTF GRanges and 
return query transcripts with different exon structures from reference 
transcripts. Transcriptome assemblers may sometime extend 5' and 3' ends 
of known transcripts based on experimental data. These annotated transcripts 
can be removed by inputting "intron" to the refine.by argument. This will 
further compare and remove transcripts of identical intron structures. 
Alternatively, transcripts with unique CDS coordinates can be selected by 
typing "cds" to the refine.by argument.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Fursham Hamid</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># Load dataset</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">matched_query_gtf</span>, <span class="va">ref_gtf</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># shortlist new transcripts</span></span>
<span class="r-in"><span class="fu">subsetNewTranscripts</span><span class="op">(</span><span class="va">matched_query_gtf</span>, <span class="va">ref_gtf</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing transcripts with exact exon coordinates</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 27 ranges and 6 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames            ranges strand |       type transcript_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;factor&gt;   &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]    chr10 79854652-79863424      + | transcript   transcript3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]    chr10 79854652-79854721      + | exon         transcript3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]    chr10 79856504-79856534      + | exon         transcript3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]    chr10 79858752-79858824      + | exon         transcript3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]    chr10 79858952-79859271      + | exon         transcript3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...               ...    ... .        ...           ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [23]    chr10 79862014-79862047      + |       exon   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [24]    chr10 79862449-79862541      + |       exon   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [25]    chr10 79862653-79862869      + |       exon   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [26]    chr10 79862978-79863055      + |       exon   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [27]    chr10 79863145-79864359      + |       exon   transcript4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      gene_id old_gene_id match_level   gene_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;character&gt; &lt;character&gt;   &lt;numeric&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...                   ...         ...         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [23] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [24] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [25] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [26] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [27] ENSMUSG00000006498.14       GeneA           4       Ptbp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

