<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using factR • factR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using factR">
<meta property="og:description" content="factR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">factR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/quickstart.html">
    <span class="fas fa-rocket-launch"></span>
     
    Quick start
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fas fa-book-open"></span>
     
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/factR.html">1) factR workflow</a>
    </li>
    <li>
      <a href="../articles/analyzing_outputs.html">2) Analyzing factR output</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper"></span>
     
    News
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fursham-h/factR" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/FurshamH" class="external-link">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="factR_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using factR</h1>
                        <h4 data-toc-skip class="author">Fursham Hamid</h4>
            
            <h4 data-toc-skip class="date">2022-08-31</h4>
      
      
      <div class="hidden name"><code>factR.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Many eukaryotic genes give rise to multiple RNA isoforms, increasing the protein-coding capacity of the genome and extending the range of post-transcriptional regulation possibilities. High-throughput sequencing is often used to deduce repertoires of transcripts expressed in specific biological samples by aligning the data to genomic sequences and assembling the alignments into transcript architectures. This typically outputs Gene Transfer Format (GTF) files describing newly identified transcripts as sets of exonic coordinates, but lacking the information about their coding sequences (CDSs; also known as ORFs) and possible biological functions.</p>
<p>To this end, we developed a package for functional annotation of custom-assembled transcriptomes in R (<em>factR</em>). <em>factR</em> predicts CDSs for novel RNA isoforms using a reference-guided process and then determines domain organisation of the protein products and possible susceptibility of transcripts to nonsense-mediated decay (NMD; a pathway destabilizing mRNAs with premature translation termination codons). <em>factR</em> also provides supporting tools for matching new transcripts to “official” gene IDs, visualizing transcript architectures and annotating alternatively spliced segments.</p>
</div>
<div class="section level2">
<h2 id="getting-started">Getting started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<div class="section level3">
<h3 id="installing-factr">Installing <em>factR</em><a class="anchor" aria-label="anchor" href="#installing-factr"></a>
</h3>
<p>To install <em>factR</em>, enter the following commands in your R environment:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"factR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="materials-needed">Materials needed<a class="anchor" aria-label="anchor" href="#materials-needed"></a>
</h3>
<div class="section level4">
<h4 id="custom-assembled-transcriptome">Custom-assembled transcriptome<a class="anchor" aria-label="anchor" href="#custom-assembled-transcriptome"></a>
</h4>
<p><em>factR</em> requires a custom transcriptome file in the GTF format as an input, and we provide the following three sample custom transcriptome files that can be used to test <em>factR</em> tools:</p>
<ol style="list-style-type: decimal">
<li>
<strong>bulk_merged_sample.gtf.gz</strong> assembled using the HISAT2-StringtTie2 pipeline <span class="citation">(Pertea et al. 2016)</span> from bulk RNA-seq data for mouse embryonic stem cells treated with the NMD inhibitor cycloheximide (CHX) or left untreated as a control.<br>
</li>
<li>
<strong>sc_merged_sample.gtf.gz</strong> assembled using the HISAT2-StringtTie2 pipeline from single-cell RNA-seq data for glutamatergic neurons, GABAergic neurons, astrocytes or endothelial cells from the mouse visual cortex <span class="citation">(Tasic et al. 2018)</span>. A simplified version of this GTF containing the first 500 genes on chr15 is available as part of <em>factR</em>.</li>
<li>
<strong>lr_merged_sample.gtf.gz</strong> assembled using minimap2 <span class="citation">(Li 2018)</span> and StringtTie2 <span class="citation">(Kovaka et al. 2019)</span> from brain-specific long-read Oxford Nanopore RNA sequencing data <span class="citation">(Sessegolo et al. 2019)</span>.</li>
</ol>
<p>Method to download the above GTF files is described in the <a href="#importing-and-inspecting-gtf-data">“Importing and inspecting GTF data”</a> section below.<br>
Users may alternatively prepare their own GTF files making sure that these contain both <strong>gene_id</strong> and <strong>transcript_id</strong> attributes.<br><!-- *In the future, add how to build custom GTF files* --></p>
</div>
<div class="section level4">
<h4 id="reference-transcriptome">Reference transcriptome<a class="anchor" aria-label="anchor" href="#reference-transcriptome"></a>
</h4>
<p>Several <em>factR</em> functions require reference transcriptome files (GTF or GFF3) as a guide. Such files can be accessed from within the R environment using e.g.  the R package <em>AnnotationHub</em> or downloaded from an external database such as GENCODE or Ensembl. Both possibilities are described in the <a href="#updating-gene-info">“Updating gene info”</a> section below.</p>
</div>
<div class="section level4">
<h4 id="genome">Genome<a class="anchor" aria-label="anchor" href="#genome"></a>
</h4>
<p><em>factR</em> needs genomic DNA sequence to predict CDSs. Users may obtain genome files using e.g. <em>BSgenome</em> or <em>AnnotationHub</em> or download them from an external database such as GENCODE or Ensembl. We describe this in more detail in the <a href="#constructing-cds-information">“Constructing CDS information”</a> section below.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="using-factr">Using <em>factR</em><a class="anchor" aria-label="anchor" href="#using-factr"></a>
</h2>
<p>Load <em>factR</em> to the R environment as follows:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://fursham-h.github.io/factR/">factR</a></span><span class="op">)</span>  </span></code></pre></div>
<div class="section level3">
<h3 id="importing-and-inspecting-gtf-data">Importing and inspecting GTF data<a class="anchor" aria-label="anchor" href="#importing-and-inspecting-gtf-data"></a>
</h3>
<p><em>factR</em> handles transcriptome information in the form of <em>GenomicRanges</em> objects containing genomic interval data and relevant metadata. To create such an object from a GTF file, we use the <code>importGTF</code> function:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sc_merged_sample.gtf.gz"</span>, package <span class="op">=</span> <span class="st">"factR"</span><span class="op">)</span></span>
<span><span class="va">custom.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/importGTF.html">importGTF</a></span><span class="op">(</span><span class="va">gtf</span><span class="op">)</span>  </span></code></pre></div>
<!-- This should import sc_merged_sample.gtf.gz, and we will use this input in the  -->
<!-- walk-through below. To try bulk_merged.gtf.gz or lr_merged.gtf.gz, simply  -->
<!-- change "sc" to "bulk" or "lr" in the selected.data variable. Users may import  -->
<!-- their own GTF file by specifying the path to such file in the `importGTF`  -->
<!-- function.  -->
<p>The imported GTF file is stored as a <em>GenomicRanges</em> object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">custom.gtf</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; [1] "GRanges"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "GenomicRanges"</span></span></code></pre></div>
<p>Contents of the object can be examined using <code>head</code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">custom.gtf</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; GRanges object with 6 ranges and 9 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames          ranges strand |    source       type     score</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;factor&gt;   &lt;factor&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]    chr15 3180731-3180944      * | StringTie transcript      1000</span></span>
<span><span class="co">#&gt;   [2]    chr15 3180731-3180944      * | StringTie exon            1000</span></span>
<span><span class="co">#&gt;   [3]    chr15 3217391-3219698      - | StringTie transcript      1000</span></span>
<span><span class="co">#&gt;   [4]    chr15 3217391-3219698      - | StringTie exon            1000</span></span>
<span><span class="co">#&gt;   [5]    chr15 3268547-3277274      + | StringTie transcript      1000</span></span>
<span><span class="co">#&gt;   [6]    chr15 3268547-3268768      + | StringTie exon            1000</span></span>
<span><span class="co">#&gt;           phase     gene_id        transcript_id exon_number   gene_name</span></span>
<span><span class="co">#&gt;       &lt;integer&gt; &lt;character&gt;          &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;   [1]      &lt;NA&gt; MSTRG.14523        MSTRG.14523.1        &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [2]      &lt;NA&gt; MSTRG.14523        MSTRG.14523.1           1        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [3]      &lt;NA&gt; MSTRG.14524 ENSMUST00000227053.1        &lt;NA&gt;      Gm7962</span></span>
<span><span class="co">#&gt;   [4]      &lt;NA&gt; MSTRG.14524 ENSMUST00000227053.1           1      Gm7962</span></span>
<span><span class="co">#&gt;   [5]      &lt;NA&gt; MSTRG.14525 ENSMUST00000160787.8        &lt;NA&gt;     Selenop</span></span>
<span><span class="co">#&gt;   [6]      &lt;NA&gt; MSTRG.14525 ENSMUST00000160787.8           1     Selenop</span></span>
<span><span class="co">#&gt;                 ref_gene_id</span></span>
<span><span class="co">#&gt;                 &lt;character&gt;</span></span>
<span><span class="co">#&gt;   [1]                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [2]                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [3]  ENSMUSG00000114999.1</span></span>
<span><span class="co">#&gt;   [4]  ENSMUSG00000114999.1</span></span>
<span><span class="co">#&gt;   [5] ENSMUSG00000064373.12</span></span>
<span><span class="co">#&gt;   [6] ENSMUSG00000064373.12</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>In addition to genomic coordinates (seqnames and ranges), a typical GTF file contains metadata describing the feature type (e.g. transcript or exon), transcript IDs and some information on their parental genes.</p>
<p>Use the following command to calculate the total number of transcripts in the input transcriptome:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">custom.gtf</span><span class="op">$</span><span class="va">transcript_id</span><span class="op">)</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; [1] 1202</span></span></code></pre></div>
<p>Note that none of the transcripts in the custom.gtf object contain CDS information:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">custom.gtf</span><span class="op">[</span><span class="va">custom.gtf</span><span class="op">$</span><span class="va">type</span><span class="op">==</span><span class="st">"CDS"</span><span class="op">]</span><span class="op">$</span><span class="va">transcript_id</span><span class="op">)</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; [1] 0</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plotting-transcript-structures">Plotting transcript structures<a class="anchor" aria-label="anchor" href="#plotting-transcript-structures"></a>
</h3>
<p>Users may visualize specific sets of transcripts using <code>viewTranscripts</code>. For example, the following will plot transcripts from the <em>Zfr</em> gene encoding a conserved zinc finger-containing RNA-binding protein with known neuronal functions :</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">custom.gtf</span>, <span class="st">"Zfr"</span><span class="op">)</span>  </span></code></pre></div>
<p><img src="factR_files/figure-html/plottranscripts-1.png" width="700"></p>
<p>StringTie2, a popular transcript assembler used to generate our custom GTF, typically assigns arbitrary names to newly identified transcripts (e.g. “MSTRG.x.y”) and uses the same prefix for their gene IDs (e.g. “MSTRG.x”). Incidentally, this is why the output above contains 10 previously known <em>Zfr</em> transcripts but lacks any novel entries.</p>
</div>
<div class="section level3">
<h3 id="updating-gene-info">Updating gene info<a class="anchor" aria-label="anchor" href="#updating-gene-info"></a>
</h3>
<p><em>factR</em> can update gene metadata in custom transcriptome objects using a reference annotation as guide. Below, we describe two alternative ways to obtain mouse reference transcriptome data from GENCODE.</p>
<ol style="list-style-type: decimal">
<li>Using <em>AnnotationHub</em> package</li>
</ol>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># query database for mouse gencode basic annotation  </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span>  </span>
<span><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; snapshotDate(): 2022-04-25</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Mus musculus'</span>, <span class="st">'gencode'</span>, <span class="st">'gff'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 9 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2022-04-25</span></span>
<span><span class="co">#&gt; # $dataprovider: Gencode</span></span>
<span><span class="co">#&gt; # $species: Mus musculus</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH49545"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             title                                                    </span></span>
<span><span class="co">#&gt;   AH49545 | gencode.vM6.2wayconspseudos.gff3.gz                      </span></span>
<span><span class="co">#&gt;   AH49546 | gencode.vM6.annotation.gff3.gz                           </span></span>
<span><span class="co">#&gt;   AH49547 | gencode.vM6.basic.annotation.gff3.gz                     </span></span>
<span><span class="co">#&gt;   AH49548 | gencode.vM6.chr_patch_hapl_scaff.annotation.gff3.gz      </span></span>
<span><span class="co">#&gt;   AH49549 | gencode.vM6.chr_patch_hapl_scaff.basic.annotation.gff3.gz</span></span>
<span><span class="co">#&gt;   AH49550 | gencode.vM6.long_noncoding_RNAs.gff3.gz                  </span></span>
<span><span class="co">#&gt;   AH49551 | gencode.vM6.polyAs.gff3.gz                               </span></span>
<span><span class="co">#&gt;   AH49552 | gencode.vM6.primary_assembly.annotation.gff3.gz          </span></span>
<span><span class="co">#&gt;   AH49553 | gencode.vM6.tRNAs.gff3.gz</span></span>
<span>  </span>
<span><span class="co"># Download full annotation  </span></span>
<span><span class="va">ref.gtf</span> <span class="op">&lt;-</span> <span class="va">ah</span><span class="op">[[</span><span class="st">'AH49546'</span><span class="op">]</span><span class="op">]</span>  </span>
<span><span class="co">#&gt; downloading 1 resources</span></span>
<span><span class="co">#&gt; retrieving 1 resource</span></span>
<span><span class="co">#&gt; loading from cache</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Downloading from a GENCODE database (ref.gtf generated from this method will be used for the remaining of the workflow)</li>
</ol>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M25/gencode.vM25.annotation.gtf.gz"</span>,  </span>
<span>              destfile <span class="op">=</span> <span class="va">tmp</span><span class="op">)</span></span>
<span><span class="va">ref.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/importGTF.html">importGTF</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span> </span></code></pre></div>
<p>When choosing a reference, users should consider using one with the same chromosome naming style (e.g. “chr1” or “1”). Alternatively, the styles can be matched using the <code>matchChromosomes</code> function (see <code><a href="../reference/matchChromosomes.html">help(matchChromosomes)</a></code> for more detail).</p>
<p>Once the ref.gtf object is ready, novel transcripts in custom.gtf can be assigned to “official” gene IDs, whenever possible, using <code>matchGeneInfo</code>. By default, this function matches the query (custom.gtf) to the reference (ref.gtf) by finding overlapping coordinates:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># matching gene metadata  </span></span>
<span><span class="va">custom_matched_1.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/matchGeneInfo.html">matchGeneInfo</a></span><span class="op">(</span><span class="va">custom.gtf</span>, <span class="va">ref.gtf</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; <span style="font-style: italic;">    Number of mismatched gene_ids found: 500</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    ---&gt; Attempting to match gene_ids by finding overlapping coordinates...</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    ---&gt; 386 gene_id matched</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    Total gene_ids corrected: 386</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    Remaining number of mismatched gene_ids: 114</span></span></span></code></pre></div>
<p>To tune the performance of <code>matchGeneInfo</code>, we provide additional arguments specifying the name of columns containing “primary” and potentially “secondary” gene IDs from the query (custom.gtf). For further information, see <code><a href="../reference/matchGeneInfo.html">help(matchGeneInfo)</a></code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># matching gene metadata  </span></span>
<span><span class="va">custom_matched_2.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/matchGeneInfo.html">matchGeneInfo</a></span><span class="op">(</span><span class="va">custom.gtf</span>, <span class="va">ref.gtf</span>,   </span>
<span>                            primary_gene_id <span class="op">=</span> <span class="st">"gene_id"</span>,   </span>
<span>                            secondary_gene_id <span class="op">=</span> <span class="st">"ref_gene_id"</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; <span style="font-style: italic;">    Number of mismatched gene_ids found: 500</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    -&gt; Attempting to correct gene ids by replacing gene_id with ref_gene_id...</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    -&gt; 212 gene_ids matched</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    --&gt; Attempting to match ensembl gene_ids...</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    --&gt; All ensembl gene ids have been matched</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    ---&gt; Attempting to match gene_ids by finding overlapping coordinates...</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    ---&gt; 174 gene_id matched</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    Total gene_ids corrected: 386</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    Remaining number of mismatched gene_ids: 114</span></span></span></code></pre></div>
<p>Note that custom.gtf updated by <code>matchGeneInfo</code> now contains 10 known and 7 newly <em>Zfr</em> assembled transcripts:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">custom_matched_2.gtf</span>, <span class="st">"Zfr"</span><span class="op">)</span>  </span></code></pre></div>
<p><img src="factR_files/figure-html/plottranscripts2-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="shortlisting-novel-transcripts">Shortlisting novel transcripts<a class="anchor" aria-label="anchor" href="#shortlisting-novel-transcripts"></a>
</h3>
<p>As seen in the above example, custom transcriptomes typically combine both new and previously annotated transcripts. To select only newly predicted transcripts, run the following:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">custom_new.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subsetNewTranscripts.html">subsetNewTranscripts</a></span><span class="op">(</span><span class="va">custom_matched_2.gtf</span>, <span class="va">ref.gtf</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; Removing transcripts with exact exon coordinates</span></span>
<span></span>
<span><span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">custom_new.gtf</span>,<span class="st">"Zfr"</span><span class="op">)</span>  </span></code></pre></div>
<p><img src="factR_files/figure-html/subsettx-1.png" width="700"></p>
<p>This will subset custom.gtf transcripts with distinct exonic coordinates compared to ref.gtf and will store these transcripts in the custom_new.gtf object. Some custom-built transcripts may differ from their reference counterparts by having different start or/and end coordinates, with otherwise similar exon-intron structure. To shortlist novel transcripts with distinct intronic coordinates only, simply set the “refine.by” argument to “intron”:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">custom_new.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subsetNewTranscripts.html">subsetNewTranscripts</a></span><span class="op">(</span><span class="va">custom_matched_2.gtf</span>, <span class="va">ref.gtf</span>, refine.by <span class="op">=</span> <span class="st">"intron"</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; Removing transcripts with exact exon coordinates</span></span>
<span><span class="co">#&gt; Removing transcripts with exact intron coordinates</span></span>
<span></span>
<span><span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">custom_new.gtf</span>, <span class="st">"Zfr"</span><span class="op">)</span>  </span></code></pre></div>
<p><img src="factR_files/figure-html/subsettx2-1.png" width="700"></p>
<p>We will use the custom_new.gtf object in the rest of the workflow.</p>
</div>
<div class="section level3">
<h3 id="constructing-cds-information">Constructing CDS information<a class="anchor" aria-label="anchor" href="#constructing-cds-information"></a>
</h3>
<p>Functional annotation of newly assembled transcripts in <em>factR</em> begins by deducing their protein-coding sequences (CDSs). To search for putative CDSs, <em>factR</em> requires a genome sequence file, which can be obtained using R packages or downloaded from online databases (e.g. UCSC, GENCODE or Ensembl). Three alternative ways to retrieve mouse genomic sequence are described below.</p>
<ol style="list-style-type: decimal">
<li>Using <em>BSgenome</em> (Mmusculus generated from this method will be used for the remaining of the workflow) This package supports most sequenced genomes. Mouse mm10 sequence can be downloaded as follows:<br><!-- Installation code below is for demonstration and not evaluated -->
</li>
</ol>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>  </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>  </span>
<span>  </span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"BSgenome.Mmusculus.UCSC.mm10"</span><span class="op">)</span>  </span></code></pre></div>
<p>and loaded into R environment:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Mmusculus.UCSC.mm10</span><span class="op">)</span></span>
<span><span class="va">Mmusculus</span> <span class="op">&lt;-</span> <span class="va">BSgenome.Mmusculus.UCSC.mm10</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Using <em>AnnotationHub</em>
</li>
</ol>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span>   </span>
<span><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; snapshotDate(): 2022-04-25</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mm10"</span>,<span class="st">"2bit"</span><span class="op">)</span><span class="op">)</span>   </span>
<span><span class="co">#&gt; AnnotationHub with 1 record</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2022-04-25</span></span>
<span><span class="co">#&gt; # names(): AH14005</span></span>
<span><span class="co">#&gt; # $dataprovider: UCSC</span></span>
<span><span class="co">#&gt; # $species: Mus musculus</span></span>
<span><span class="co">#&gt; # $rdataclass: TwoBitFile</span></span>
<span><span class="co">#&gt; # $rdatadateadded: 2014-12-15</span></span>
<span><span class="co">#&gt; # $title: mm10.2bit</span></span>
<span><span class="co">#&gt; # $description: UCSC 2 bit file for mm10 </span></span>
<span><span class="co">#&gt; # $taxonomyid: 10090</span></span>
<span><span class="co">#&gt; # $genome: mm10</span></span>
<span><span class="co">#&gt; # $sourcetype: TwoBit</span></span>
<span><span class="co">#&gt; # $sourceurl: http://hgdownload.cse.ucsc.edu/goldenpath/mm10/bigZips/mm10.2bit</span></span>
<span><span class="co">#&gt; # $sourcesize: NA</span></span>
<span><span class="co">#&gt; # $tags: c("2bit", "UCSC", "genome") </span></span>
<span><span class="co">#&gt; # retrieve record with 'object[["AH14005"]]'</span></span></code></pre></div>
<!-- Code below not evaluated as it serves as an alternative method of retrieving genome data -->
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Retrieve mouse genome</span></span>
<span><span class="va">Mmusculus</span> <span class="op">&lt;-</span> <span class="va">ah</span><span class="op">[[</span><span class="st">'AH14005'</span><span class="op">]</span><span class="op">]</span>  </span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Downloading from a database (e.g. GENCODE)<br><!-- Code below not evaluated as it serves as an alternative method of retrieving genome data -->
</li>
</ol>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M25/GRCm38.primary_assembly.genome.fa.gz"</span>,  </span>
<span>              <span class="va">tmp</span><span class="op">)</span>  </span>
<span><span class="va">Mmusculus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/importFASTA.html">importFASTA</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">tmp</span>, <span class="st">"/GRCm38.primary_assembly.genome.fa.gz"</span><span class="op">)</span><span class="op">)</span>  </span></code></pre></div>
<p>Once the genome sequence object is ready, <em>factR</em> can predict CDSs using its buildCDS() function and reference transcriptome data as a guide. buildCDS() first generates a database of previously annotated ATGs and uses this information to search for a potential translation start sites in query transcripts. buildCDS() then deduces the CDS and appends its coordinates to the custom transcriptome object. Let’s run this function for our novel transcripts:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">custom_new_CDS.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/buildCDS.html">buildCDS</a></span><span class="op">(</span><span class="va">custom_new.gtf</span>, <span class="va">ref.gtf</span>, <span class="va">Mmusculus</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; <span style="font-style: italic;">    Searching for reference mRNAs in query</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    No reference mRNAs found</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    Building database of annotated ATG codons</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    Selecting best ATG start codon for remaining transcripts and determining open-reading frame</span></span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">    114 new CDSs constructed</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co"><span style="font-style: italic;">#&gt;     Summary: Out of 380 transcripts in `custom_new.gtf`,</span></span></span>
<span><span class="co"><span style="font-style: italic;">#&gt;     114 transcript CDSs were built</span></span></span></code></pre></div>
<p>Note that the novel <em>Zfr</em> transcripts have been updated with information about likely CDSs (dark blue) and untranslated regions (light blue) :</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="st">"Zfr"</span><span class="op">)</span>  </span></code></pre></div>
<p><img src="factR_files/figure-html/viewafterCDS-1.png" width="700"></p>
<p>We can display exonic regions and CDSs more clearly (at the expense of loosing their bona fide genomic coordinates) by setting rescale_intron argument to TRUE.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="st">"Zfr"</span>, rescale_introns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>  </span></code></pre></div>
<p><img src="factR_files/figure-html/viewafterCDSscale-1.png" width="700"></p>
<p>For comparison, here is the CDS situation in the reference Zfr transcripts:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">ref.gtf</span>, <span class="st">"Zfr"</span>, rescale_introns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>  </span></code></pre></div>
<p><img src="factR_files/figure-html/viewrefCDS-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="predicting-nmd">Predicting NMD<a class="anchor" aria-label="anchor" href="#predicting-nmd"></a>
</h3>
<p>To explore possible susceptibility of newly identified mRNA isoforms to NMD, we use the <code>predictNMD</code> function:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NMDprediction.out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictNMD.html">predictNMD</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; <span style="font-style: italic;"> Predicting NMD sensitivities for 114 mRNAs</span></span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">NMDprediction.out</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 5</span></span></span>
<span><span class="co">#&gt;   transcript           stop_to_lastEJ num_of_downEJs `3'UTR_length` is_NMD</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ENSMUST00000100790.3            222              3           <span style="text-decoration: underline;">2</span>895 TRUE  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ENSMUST00000228827.1             <span style="color: #BB0000;">NA</span>              0           <span style="text-decoration: underline;">1</span>672 FALSE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> MSTRG.14527.7                  -<span style="color: #BB0000;">131</span>              0           <span style="text-decoration: underline;">2</span>244 FALSE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> MSTRG.14531.1                  <span style="text-decoration: underline;">1</span>937              9           <span style="text-decoration: underline;">4</span>141 TRUE  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> MSTRG.14531.5                   872              3           <span style="text-decoration: underline;">5</span>317 TRUE  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> MSTRG.14533.2                  <span style="text-decoration: underline;">1</span>073              1           <span style="text-decoration: underline;">2</span>541 TRUE</span></span></code></pre></div>
<p><code>predictNMD</code> outputs a data frame containing information on features that may promote NMD and predicts NMD sensitivity for each CDS-containing transcript based on the distance between the stop codon and the last exon-exon junction.</p>
<p>To identify putative NMD targets for specific genes (e.g. <em>Zfr</em>), run the following:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NMDprediction.Zfr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictNMD.html">predictNMD</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span><span class="op">)</span> </span>
<span><span class="co">#&gt; <span style="font-style: italic;"> Predicting NMD sensitivities for 7 mRNAs</span></span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">NMDprediction.Zfr</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 5</span></span></span>
<span><span class="co">#&gt;   transcript     stop_to_lastEJ num_of_downEJs `3'UTR_length` is_NMD</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> MSTRG.14649.1            <span style="text-decoration: underline;">3</span>857             11           <span style="text-decoration: underline;">8</span>774 TRUE  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> MSTRG.14649.10           -<span style="color: #BB0000;">177</span>              0           <span style="text-decoration: underline;">4</span>740 FALSE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> MSTRG.14649.11             41              1           <span style="text-decoration: underline;">4</span>958 FALSE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> MSTRG.14649.12           -<span style="color: #BB0000;">300</span>              0           <span style="text-decoration: underline;">2</span>242 FALSE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> MSTRG.14649.3            <span style="text-decoration: underline;">3</span>857             11           <span style="text-decoration: underline;">8</span>774 TRUE  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> MSTRG.14649.4            -<span style="color: #BB0000;">300</span>              0           <span style="text-decoration: underline;">2</span>242 FALSE</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="predicting-protein-domains">Predicting protein domains<a class="anchor" aria-label="anchor" href="#predicting-protein-domains"></a>
</h3>
<p><em>factR</em> can also inspect domain structure of protein products encoded by newly identified mRNA isoforms using its <code><a href="../reference/predictDomains.html">predictDomains()</a></code> function. This requires connection to the online PFAM database and may require a substantial amount of time and stable internet connection to query multiple transcripts simultaneously. To quickly explore functionality of the <code><a href="../reference/predictDomains.html">predictDomains()</a></code> tool, users may prefer to begin with a relatively small subset of transcripts. For example, the following will predict domain structure for all transcripts of the Zfr gene:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/predictDomains.html">predictDomains</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="va">Mmusculus</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span>, progress_bar <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; Checking CDSs and translating protein sequences</span></span>
<span><span class="co">#&gt; Predicting domain families for 7 proteins</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 5</span></span></span>
<span><span class="co">#&gt;   transcript    description                  eval     begin   end</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> MSTRG.14649.1 beta-beta-alpha zinc fingers 1.54e-20   303   364</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> MSTRG.14649.1 beta-beta-alpha zinc fingers 1.54e-20   371   414</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> MSTRG.14649.3 beta-beta-alpha zinc fingers 1.54e-20   303   350</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> MSTRG.14649.4 beta-beta-alpha zinc fingers 1.54e-20   311   358</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> MSTRG.14649.9 beta-beta-alpha zinc fingers 1.54e-20   289   350</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> MSTRG.14649.9 beta-beta-alpha zinc fingers 1.54e-20   357   400</span></span></code></pre></div>
<p>The domain architectures for the set of query proteins can be additionally plotted by switching the argument “plot” to TRUE:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">domains.Zfr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictDomains.html">predictDomains</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="va">Mmusculus</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span>, plot <span class="op">=</span> <span class="cn">TRUE</span>, progress_bar <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>  </span></code></pre></div>
<p>If you want to predict domains for all new transcripts and do not mind waiting for some time depending on the connection speed and the PFAM server load, run the following:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">domains.out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictDomains.html">predictDomains</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="va">Mmusculus</span>, progress_bar <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>  </span></code></pre></div>
</div>
<div class="section level3">
<h3 id="export-output-objects">Export output objects<a class="anchor" aria-label="anchor" href="#export-output-objects"></a>
</h3>
<p>Annotated <em>custom_new_CDS.gtf</em> object can be exported to a GTF file as follows:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span> </span>
<span><span class="fu">export</span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="st">"Custom_new.gtf"</span>, format <span class="op">=</span> <span class="st">"GTF"</span><span class="op">)</span>  </span></code></pre></div>
<p>Finally, to export <em>NMDprediction.out</em> and <em>domains.out</em> data frames as tab-delimited text files, run the following:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">NMDprediction.out</span>, <span class="st">"Custom_new_NMD.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, quote <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>  </span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">domains.out</span>, <span class="st">"Custom_new_domains.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, quote <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>  </span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="citing-factr">Citing <em>factR</em><a class="anchor" aria-label="anchor" href="#citing-factr"></a>
</h2>
<p>Please cite <em>factR</em> if you find it useful:<br><!-- Citation will be added in the near future --></p>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This workflow was conducted on:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] forcats_0.5.2                      stringr_1.4.1                     </span></span>
<span><span class="co">#&gt;  [3] dplyr_1.0.9                        purrr_0.3.4                       </span></span>
<span><span class="co">#&gt;  [5] readr_2.1.2                        tidyr_1.2.0                       </span></span>
<span><span class="co">#&gt;  [7] tibble_3.1.8                       ggplot2_3.3.6                     </span></span>
<span><span class="co">#&gt;  [9] tidyverse_1.3.2                    GenomicFeatures_1.48.3            </span></span>
<span><span class="co">#&gt; [11] AnnotationDbi_1.58.0               Biobase_2.56.0                    </span></span>
<span><span class="co">#&gt; [13] BSgenome.Mmusculus.UCSC.mm10_1.4.3 BSgenome_1.64.0                   </span></span>
<span><span class="co">#&gt; [15] rtracklayer_1.56.1                 GenomicRanges_1.48.0              </span></span>
<span><span class="co">#&gt; [17] Biostrings_2.64.1                  GenomeInfoDb_1.32.3               </span></span>
<span><span class="co">#&gt; [19] XVector_0.36.0                     IRanges_2.30.1                    </span></span>
<span><span class="co">#&gt; [21] S4Vectors_0.34.0                   AnnotationHub_3.4.0               </span></span>
<span><span class="co">#&gt; [23] BiocFileCache_2.4.0                dbplyr_2.2.1                      </span></span>
<span><span class="co">#&gt; [25] BiocGenerics_0.42.0                factR_0.99.9                      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] readxl_1.4.1                  backports_1.4.1              </span></span>
<span><span class="co">#&gt;   [3] systemfonts_1.0.4             BiocParallel_1.30.3          </span></span>
<span><span class="co">#&gt;   [5] digest_0.6.29                 htmltools_0.5.3              </span></span>
<span><span class="co">#&gt;   [7] fansi_1.0.3                   magrittr_2.0.3               </span></span>
<span><span class="co">#&gt;   [9] memoise_2.0.1                 googlesheets4_1.0.1          </span></span>
<span><span class="co">#&gt;  [11] tzdb_0.3.0                    modelr_0.1.9                 </span></span>
<span><span class="co">#&gt;  [13] matrixStats_0.62.0            pkgdown_2.0.6                </span></span>
<span><span class="co">#&gt;  [15] prettyunits_1.1.1             colorspace_2.0-3             </span></span>
<span><span class="co">#&gt;  [17] blob_1.2.3                    rvest_1.0.3                  </span></span>
<span><span class="co">#&gt;  [19] rappdirs_0.3.3                textshaping_0.3.6            </span></span>
<span><span class="co">#&gt;  [21] haven_2.5.1                   xfun_0.32                    </span></span>
<span><span class="co">#&gt;  [23] crayon_1.5.1                  RCurl_1.98-1.8               </span></span>
<span><span class="co">#&gt;  [25] jsonlite_1.8.0                glue_1.6.2                   </span></span>
<span><span class="co">#&gt;  [27] gtable_0.3.0                  gargle_1.2.0                 </span></span>
<span><span class="co">#&gt;  [29] zlibbioc_1.42.0               DelayedArray_0.22.0          </span></span>
<span><span class="co">#&gt;  [31] scales_1.2.1                  DBI_1.1.3                    </span></span>
<span><span class="co">#&gt;  [33] Rcpp_1.0.9                    xtable_1.8-4                 </span></span>
<span><span class="co">#&gt;  [35] progress_1.2.2                bit_4.0.4                    </span></span>
<span><span class="co">#&gt;  [37] httr_1.4.4                    ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt;  [39] pkgconfig_2.0.3               XML_3.99-0.10                </span></span>
<span><span class="co">#&gt;  [41] farver_2.1.1                  sass_0.4.2                   </span></span>
<span><span class="co">#&gt;  [43] utf8_1.2.2                    tidyselect_1.1.2             </span></span>
<span><span class="co">#&gt;  [45] labeling_0.4.2                rlang_1.0.4                  </span></span>
<span><span class="co">#&gt;  [47] later_1.3.0                   munsell_0.5.0                </span></span>
<span><span class="co">#&gt;  [49] BiocVersion_3.15.2            cellranger_1.1.0             </span></span>
<span><span class="co">#&gt;  [51] tools_4.2.1                   cachem_1.0.6                 </span></span>
<span><span class="co">#&gt;  [53] cli_3.3.0                     generics_0.1.3               </span></span>
<span><span class="co">#&gt;  [55] RSQLite_2.2.16                broom_1.0.1                  </span></span>
<span><span class="co">#&gt;  [57] evaluate_0.16                 fastmap_1.1.0                </span></span>
<span><span class="co">#&gt;  [59] yaml_2.3.5                    ragg_1.2.2                   </span></span>
<span><span class="co">#&gt;  [61] knitr_1.40                    bit64_4.0.5                  </span></span>
<span><span class="co">#&gt;  [63] fs_1.5.2                      KEGGREST_1.36.3              </span></span>
<span><span class="co">#&gt;  [65] pbapply_1.5-0                 mime_0.12                    </span></span>
<span><span class="co">#&gt;  [67] xml2_1.3.3                    biomaRt_2.52.0               </span></span>
<span><span class="co">#&gt;  [69] compiler_4.2.1                filelock_1.0.2               </span></span>
<span><span class="co">#&gt;  [71] curl_4.3.2                    png_0.1-7                    </span></span>
<span><span class="co">#&gt;  [73] interactiveDisplayBase_1.34.0 reprex_2.0.2                 </span></span>
<span><span class="co">#&gt;  [75] bslib_0.4.0                   stringi_1.7.8                </span></span>
<span><span class="co">#&gt;  [77] highr_0.9                     desc_1.4.1                   </span></span>
<span><span class="co">#&gt;  [79] lattice_0.20-45               Matrix_1.4-1                 </span></span>
<span><span class="co">#&gt;  [81] vctrs_0.4.1                   pillar_1.8.1                 </span></span>
<span><span class="co">#&gt;  [83] lifecycle_1.0.1               BiocManager_1.30.18          </span></span>
<span><span class="co">#&gt;  [85] jquerylib_0.1.4               data.table_1.14.2            </span></span>
<span><span class="co">#&gt;  [87] bitops_1.0-7                  wiggleplotr_1.20.0           </span></span>
<span><span class="co">#&gt;  [89] httpuv_1.6.5                  patchwork_1.1.2              </span></span>
<span><span class="co">#&gt;  [91] R6_2.5.1                      BiocIO_1.6.0                 </span></span>
<span><span class="co">#&gt;  [93] promises_1.2.0.1              codetools_0.2-18             </span></span>
<span><span class="co">#&gt;  [95] assertthat_0.2.1              SummarizedExperiment_1.26.1  </span></span>
<span><span class="co">#&gt;  [97] rprojroot_2.0.3               rjson_0.2.21                 </span></span>
<span><span class="co">#&gt;  [99] withr_2.5.0                   GenomicAlignments_1.32.1     </span></span>
<span><span class="co">#&gt;  [ reached getOption("max.print") -- omitted 11 entries ]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references">
<div id="ref-Kovaka2019">
<p>Kovaka, Sam, Aleksey V. Zimin, Geo M. Pertea, Roham Razaghi, Steven L. Salzberg, and Mihaela Pertea. 2019. “Transcriptome Assembly from Long-Read Rna-Seq Alignments with Stringtie2.” <em>Genome Biology</em> 20 (1): 278. <a href="https://doi.org/10.1186/s13059-019-1910-1" class="external-link">https://doi.org/10.1186/s13059-019-1910-1</a>.</p>
</div>
<div id="ref-Li2018">
<p>Li, Heng. 2018. “Minimap2: Pairwise Alignment for Nucleotide Sequences.” <em>Bioinformatics</em> 34 (18): 3094–3100. <a href="https://doi.org/10.1093/bioinformatics/bty191" class="external-link">https://doi.org/10.1093/bioinformatics/bty191</a>.</p>
</div>
<div id="ref-Pertea2016">
<p>Pertea, Mihaela, Daehwan Kim, Geo M. Pertea, Jeffrey T. Leek, and Steven L. Salzberg. 2016. “Transcript-Level Expression Analysis of Rna-Seq Experiments with Hisat, Stringtie and Ballgown.” <em>Nature Protocols</em> 11 (9): 1650–67. <a href="https://doi.org/10.1038/nprot.2016.095" class="external-link">https://doi.org/10.1038/nprot.2016.095</a>.</p>
</div>
<div id="ref-Sessegolo2019">
<p>Sessegolo, Camille, Corinne Cruaud, Corinne Da Silva, Audric Cologne, Marion Dubarry, Thomas Derrien, Vincent Lacroix, and Jean-Marc Aury. 2019. “Transcriptome Profiling of Mouse Samples Using Nanopore Sequencing of cDNA and Rna Molecules.” <em>Scientific Reports</em> 9 (1): 14908. <a href="https://doi.org/10.1038/s41598-019-51470-9" class="external-link">https://doi.org/10.1038/s41598-019-51470-9</a>.</p>
</div>
<div id="ref-Tasic2018">
<p>Tasic, Bosiljka, Zizhen Yao, Lucas T. Graybuck, Kimberly A. Smith, Thuc Nghi Nguyen, Darren Bertagnolli, Jeff Goldy, et al. 2018. “Shared and Distinct Transcriptomic Cell Types Across Neocortical Areas.” <em>Nature</em> 563 (7729): 72–78. <a href="https://doi.org/10.1038/s41586-018-0654-5" class="external-link">https://doi.org/10.1038/s41586-018-0654-5</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
