<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using factR • factR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using factR">
<meta property="og:description" content="factR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">factR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/factR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="factR_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using factR</h1>
                        <h4 class="author">Fursham Hamid</h4>
            
            <h4 class="date">2021-01-08</h4>
      
      
      <div class="hidden name"><code>factR.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Many eukaryotic genes give rise to multiple RNA isoforms, increasing the protein-coding capacity of the genome and extending the range of post-transcriptional regulation possibilities. High-throughput sequencing is often used to deduce repertoires of transcripts expressed in specific biological samples by aligning the data to genomic sequences and assembling the alignments into transcript architectures. This typically outputs Gene Transfer Format (GTF) files describing newly identified transcripts as sets of exonic coordinates, but lacking the information about their coding sequences (CDSs; also known as ORFs) and possible biological functions.</p>
<p>To this end, we developed a package for functional annotation of custom-assembled transcriptomes in R (<em>factR</em>). <em>factR</em> predicts CDSs for novel RNA isoforms using a reference-guided process and then determines domain organisation of the protein products and possible susceptibility of transcripts to nonsense-mediated decay (NMD; a pathway destabilizing mRNAs with premature translation termination codons). <em>factR</em> also provides supporting tools for matching new transcripts to “official” gene IDs, visualizing transcript architectures and annotating alternatively spliced segments.</p>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<div id="installing-factr" class="section level2">
<h2 class="hasAnchor">
<a href="#installing-factr" class="anchor"></a>Installing <em>factR</em>
</h2>
<p>To install <em>factR</em>, enter the following commands in your R environment:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>  
<span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"fursham-h/factR"</span><span class="op">)</span>  </code></pre></div>
</div>
<div id="materials-needed" class="section level2">
<h2 class="hasAnchor">
<a href="#materials-needed" class="anchor"></a>Materials needed</h2>
<div id="custom-assembled-transcriptome" class="section level3">
<h3 class="hasAnchor">
<a href="#custom-assembled-transcriptome" class="anchor"></a>Custom-assembled transcriptome</h3>
<p><em>factR</em> requires a custom transcriptome file in the GTF format as an input, and we provide the following three custom transcriptome files that can be used to test <em>factR</em> tools:</p>
<ol style="list-style-type: decimal">
<li>
<strong>bulk_merged.gtf.gz</strong> assembled using the HISAT2-StringtTie2 pipeline <span class="citation">(Pertea et al. 2016)</span> from bulk RNA-seq data for mouse embryonic stem cells treated with the NMD inhibitor cycloheximide (CHX) or left untreated as a control.<br>
</li>
<li>
<strong>sc_merged.gtf.gz</strong> assembled using the HISAT2-StringtTie2 pipeline from single-cell RNA-seq data for glutamatergic neurons, GABAergic neurons, astrocytes or endothelial cells from the mouse visual cortex <span class="citation">(Tasic et al. 2018)</span>.<br>
</li>
<li>
<strong>lr_merged.gtf.gz</strong> assembled using minimap2 <span class="citation">(Li 2018)</span> and StringtTie2 <span class="citation">(Kovaka et al. 2019)</span> from brain-specific long-read Oxford Nanopore RNA sequencing data <span class="citation">(Sessegolo et al. 2019)</span>.</li>
</ol>
<p>Method to download the above GTF files is described in the <a href="#importing-and-inspecting-gtf-data">“Importing and inspecting GTF data”</a> section below.<br>
Users may alternatively prepare their own GTF files making sure that these contain both <strong>gene_id</strong> and <strong>transcript_id</strong> attributes.<br><!-- *In the future, add how to build custom GTF files* --></p>
</div>
<div id="reference-transcriptome" class="section level3">
<h3 class="hasAnchor">
<a href="#reference-transcriptome" class="anchor"></a>Reference transcriptome</h3>
<p>Several <em>factR</em> functions require reference transcriptome files (GTF or GFF3) as a guide. Such files can be accessed from within the R environment using e.g. the R package <em>AnnotationHub</em> or downloaded from an external database such as GENCODE or Ensembl. Both possibilities are described in the <a href="#updating-gene-info">“Updating gene info”</a> section below.</p>
</div>
<div id="genome" class="section level3">
<h3 class="hasAnchor">
<a href="#genome" class="anchor"></a>Genome</h3>
<p><em>factR</em> needs genomic DNA sequence to predict CDSs. Users may obtain genome files using e.g. <em>BSgenome</em> or <em>AnnotationHub</em> or download them from an external database such as GENCODE or Ensembl. We describe this in more detail in the <a href="#constructing-CDS-information">“Constructing CDS information”</a> section below.</p>
</div>
</div>
</div>
<div id="using-factr" class="section level1">
<h1 class="hasAnchor">
<a href="#using-factr" class="anchor"></a>Using <em>factR</em>
</h1>
<p>Load <em>factR</em> to the R environment as follows:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">factR</span><span class="op">)</span>  </code></pre></div>
<div id="importing-and-inspecting-gtf-data" class="section level2">
<h2 class="hasAnchor">
<a href="#importing-and-inspecting-gtf-data" class="anchor"></a>Importing and inspecting GTF data</h2>
<p><em>factR</em> handles transcriptome information in the form of <em>GenomicRanges</em> objects containing genomic interval data and relevant metadata. To create such an object from a GTF file, we use the <code>importGTF</code> function:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">selected.data</span> <span class="op">&lt;-</span> <span class="st">"sc"</span>  
  
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"https://github.com/fursham-h/MiMB-factR-supplement/raw/master/"</span>,  
                     <span class="st">"Custom%20transcriptome/"</span>, <span class="va">selected.data</span>, <span class="st">"_merged.gtf.gz"</span><span class="op">)</span>,   
              <span class="st">"input.gtf.gz"</span><span class="op">)</span>  
<span class="va">custom.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/importGTF.html">importGTF</a></span><span class="op">(</span><span class="st">"input.gtf.gz"</span><span class="op">)</span>  </code></pre></div>
<p>This should import sc_merged.gtf.gz, and we will use this input in the walk-through below. To try bulk_merged.gtf.gz or lr_merged.gtf.gz, simply change “sc” to “bulk” or “lr” in the selected.data variable. Users may import your own GTF file by specifying the path to such file in the <code>importGTF</code> function.</p>
<p>The imported GTF file is stored as a <em>GenomicRanges</em> object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">custom.gtf</span><span class="op">)</span>  
<span class="co">#&gt; [1] "GRanges"</span>
<span class="co">#&gt; attr(,"package")</span>
<span class="co">#&gt; [1] "GenomicRanges"</span></code></pre></div>
<p>Contents of the object can be examined using <code>head</code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">custom.gtf</span><span class="op">)</span>  
<span class="co">#&gt; GRanges object with 6 ranges and 9 metadata columns:</span>
<span class="co">#&gt;       seqnames          ranges strand |    source       type     score</span>
<span class="co">#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;factor&gt;   &lt;factor&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1 3188551-3672854      - | StringTie transcript      1000</span>
<span class="co">#&gt;   [2]     chr1 3188551-3207317      - | StringTie exon            1000</span>
<span class="co">#&gt;   [3]     chr1 3213439-3216968      - | StringTie exon            1000</span>
<span class="co">#&gt;   [4]     chr1 3421702-3421901      - | StringTie exon            1000</span>
<span class="co">#&gt;   [5]     chr1 3670552-3672854      - | StringTie exon            1000</span>
<span class="co">#&gt;   [6]     chr1 3205901-3216968      - | StringTie transcript      1000</span>
<span class="co">#&gt;           phase     gene_id        transcript_id exon_number   gene_name</span>
<span class="co">#&gt;       &lt;integer&gt; &lt;character&gt;          &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;   [1]      &lt;NA&gt;     MSTRG.1            MSTRG.1.1        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">#&gt;   [2]      &lt;NA&gt;     MSTRG.1            MSTRG.1.1           1        &lt;NA&gt;</span>
<span class="co">#&gt;   [3]      &lt;NA&gt;     MSTRG.1            MSTRG.1.1           2        &lt;NA&gt;</span>
<span class="co">#&gt;   [4]      &lt;NA&gt;     MSTRG.1            MSTRG.1.1           3        &lt;NA&gt;</span>
<span class="co">#&gt;   [5]      &lt;NA&gt;     MSTRG.1            MSTRG.1.1           4        &lt;NA&gt;</span>
<span class="co">#&gt;   [6]      &lt;NA&gt;     MSTRG.1 ENSMUST00000162897.1        &lt;NA&gt;        Xkr4</span>
<span class="co">#&gt;                ref_gene_id</span>
<span class="co">#&gt;                &lt;character&gt;</span>
<span class="co">#&gt;   [1]                 &lt;NA&gt;</span>
<span class="co">#&gt;   [2]                 &lt;NA&gt;</span>
<span class="co">#&gt;   [3]                 &lt;NA&gt;</span>
<span class="co">#&gt;   [4]                 &lt;NA&gt;</span>
<span class="co">#&gt;   [5]                 &lt;NA&gt;</span>
<span class="co">#&gt;   [6] ENSMUSG00000051951.5</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 39 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>In addition to genomic coordinates (seqnames and ranges), a typical GTF file contains metadata describing the feature type (e.g. transcript or exon), transcript IDs and some information on their parental genes.</p>
<p>Use the following command to calculate the total number of transcripts in the input transcriptome:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">custom.gtf</span><span class="op">$</span><span class="va">transcript_id</span><span class="op">)</span><span class="op">)</span>  
<span class="co">#&gt; [1] 178195</span></code></pre></div>
<p>Note that none of the transcripts in the custom.gtf object contain CDS information:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">custom.gtf</span><span class="op">[</span><span class="va">custom.gtf</span><span class="op">$</span><span class="va">type</span><span class="op">==</span><span class="st">"CDS"</span><span class="op">]</span><span class="op">$</span><span class="va">transcript_id</span><span class="op">)</span><span class="op">)</span>  
<span class="co">#&gt; [1] 0</span></code></pre></div>
</div>
<div id="plotting-transcript-structures" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-transcript-structures" class="anchor"></a>Plotting transcript structures</h2>
<p>Users may visualize specific sets of transcripts using <code>viewTranscripts</code>. For example, the following will plot transcripts from the <em>Zfr</em> gene encoding a conserved zinc finger-containing RNA-binding protein with known neuronal functions :</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">custom.gtf</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span><span class="op">)</span>  </code></pre></div>
<p><img src="factR_files/figure-html/plottranscripts-1.png" width="700"></p>
<p>StringTie2, a popular transcript assembler used to generate our custom GTF, typically assigns arbitrary names to newly identified transcripts (e.g. “MSTRG.x.y”) and uses the same prefix for their gene IDs (e.g. “MSTRG.x”). Incidentally, this is why the output above contains 10 previously known <em>Zfr</em> transcripts but lacks any novel entries.</p>
</div>
<div id="updating-gene-info" class="section level2">
<h2 class="hasAnchor">
<a href="#updating-gene-info" class="anchor"></a>Updating gene info</h2>
<p><em>factR</em> can update gene metadata in custom transcriptome objects using a reference annotation as guide. Below, we describe two alternative ways to obtain mouse reference transcriptome data from GENCODE.</p>
<ol style="list-style-type: decimal">
<li>Using <em>AnnotationHub</em> package</li>
</ol>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># query database for mouse gencode basic annotation  </span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span>  
<span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span>  
<span class="co">#&gt; using temporary cache /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmptZ90WN/BiocFileCache</span>
<span class="co">#&gt; snapshotDate(): 2020-10-27</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Mus musculus'</span>, <span class="st">'gencode'</span>, <span class="st">'gff'</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; AnnotationHub with 9 records</span>
<span class="co">#&gt; # snapshotDate(): 2020-10-27</span>
<span class="co">#&gt; # $dataprovider: Gencode</span>
<span class="co">#&gt; # $species: Mus musculus</span>
<span class="co">#&gt; # $rdataclass: GRanges</span>
<span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span>
<span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span>
<span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span>
<span class="co">#&gt; # retrieve records with, e.g., 'object[["AH49545"]]' </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;             title                                                    </span>
<span class="co">#&gt;   AH49545 | gencode.vM6.2wayconspseudos.gff3.gz                      </span>
<span class="co">#&gt;   AH49546 | gencode.vM6.annotation.gff3.gz                           </span>
<span class="co">#&gt;   AH49547 | gencode.vM6.basic.annotation.gff3.gz                     </span>
<span class="co">#&gt;   AH49548 | gencode.vM6.chr_patch_hapl_scaff.annotation.gff3.gz      </span>
<span class="co">#&gt;   AH49549 | gencode.vM6.chr_patch_hapl_scaff.basic.annotation.gff3.gz</span>
<span class="co">#&gt;   AH49550 | gencode.vM6.long_noncoding_RNAs.gff3.gz                  </span>
<span class="co">#&gt;   AH49551 | gencode.vM6.polyAs.gff3.gz                               </span>
<span class="co">#&gt;   AH49552 | gencode.vM6.primary_assembly.annotation.gff3.gz          </span>
<span class="co">#&gt;   AH49553 | gencode.vM6.tRNAs.gff3.gz</span>
  
<span class="co"># Download full annotation  </span>
<span class="va">ref.gtf</span> <span class="op">&lt;-</span> <span class="va">ah</span><span class="op">[[</span><span class="st">'AH49546'</span><span class="op">]</span><span class="op">]</span>  
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Downloading from a database (e.g. GENCODE)</li>
</ol>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="st">"ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M25/gencode.vM25.annotation.gtf.gz"</span>,  
              destfile <span class="op">=</span> <span class="st">"gencode.vM25.annotation.gtf.gz"</span><span class="op">)</span>  
<span class="va">ref.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/importGTF.html">importGTF</a></span><span class="op">(</span><span class="st">"gencode.vM25.annotation.gtf.gz"</span><span class="op">)</span>  </code></pre></div>
<p>When choosing a reference, users should consider using one with the same chromosome naming style (e.g. “chr1” or “1”). Alternatively, the styles can be matched using the <code>matchChromosomes</code> function (see <code><a href="../reference/matchChromosomes.html">help(matchChromosomes)</a></code> for more detail).</p>
<p>Once the ref.gtf object is ready, novel transcripts in custom.gtf can be assigned to “official” gene IDs, whenever possible, using <code>matchGeneInfo</code>. By default, this function matches the query (custom.gtf) to the reference (ref.gtf) by finding overlapping coordinates:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># matching gene metadata  </span>
<span class="va">custom_matched_1.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/matchGeneInfo.html">matchGeneInfo</a></span><span class="op">(</span><span class="va">custom.gtf</span>, <span class="va">ref.gtf</span><span class="op">)</span>  
<span class="co">#&gt; Number of mismatched gene_ids found: 47815</span>
<span class="co">#&gt; ---&gt; Attempting to match gene_ids by finding overlapping coordinates...</span>
<span class="co">#&gt; ---&gt; 38455 gene_id matched</span>
<span class="co">#&gt; Total gene_ids corrected: 38455</span>
<span class="co">#&gt; Remaining number of mismatched gene_ids: 9360</span></code></pre></div>
<p>To tune the performance of <code>matchGeneInfo</code>, we provide additional arguments specifying the name of columns containing “primary” and potentially “secondary” gene IDs from the query (custom.gtf). For further information, see <code><a href="../reference/matchGeneInfo.html">help(matchGeneInfo)</a></code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># matching gene metadata  </span>
<span class="va">custom_matched_2.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/matchGeneInfo.html">matchGeneInfo</a></span><span class="op">(</span><span class="va">custom.gtf</span>, <span class="va">ref.gtf</span>,   
                            primary_gene_id <span class="op">=</span> <span class="st">"gene_id"</span>,   
                            secondary_gene_id <span class="op">=</span> <span class="st">"ref_gene_id"</span><span class="op">)</span>  
<span class="co">#&gt; Number of mismatched gene_ids found: 47815</span>
<span class="co">#&gt; -&gt; Attempting to correct gene ids by replacing gene_id with ref_gene_id...</span>
<span class="co">#&gt; -&gt; 20931 gene_ids matched</span>
<span class="co">#&gt; --&gt; Attempting to match ensembl gene_ids...</span>
<span class="co">#&gt; --&gt; All ensembl gene ids have been matched</span>
<span class="co">#&gt; ---&gt; Attempting to match gene_ids by finding overlapping coordinates...</span>
<span class="co">#&gt; ---&gt; 17524 gene_id matched</span>
<span class="co">#&gt; Total gene_ids corrected: 38455</span>
<span class="co">#&gt; Remaining number of mismatched gene_ids: 9360</span></code></pre></div>
<p>Note that custom.gtf updated by <code>matchGeneInfo</code> now contains 10 known and 7 newly <em>Zfr</em> assembled transcripts:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">custom_matched_2.gtf</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span><span class="op">)</span>  </code></pre></div>
<p><img src="factR_files/figure-html/plottranscripts2-1.png" width="700"></p>
</div>
<div id="shortlisting-novel-transcripts" class="section level2">
<h2 class="hasAnchor">
<a href="#shortlisting-novel-transcripts" class="anchor"></a>Shortlisting novel transcripts</h2>
<p>As seen in the above example, custom transcriptomes typically combine both new and previously annotated transcripts. To select only newly predicted transcripts, run the following:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">custom_new.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subsetNewTranscripts.html">subsetNewTranscripts</a></span><span class="op">(</span><span class="va">custom_matched_2.gtf</span>, <span class="va">ref.gtf</span><span class="op">)</span>  
<span class="co">#&gt; Removing transcripts with exact exon coordinates</span>

<span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">custom_new.gtf</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span><span class="op">)</span>  </code></pre></div>
<p><img src="factR_files/figure-html/subsettx-1.png" width="700"></p>
<p>This will subset custom.gtf transcripts with distinct exonic coordinates compared to ref.gtf and will store these transcripts in the custom_new.gtf object. Some custom-built transcripts may differ from their reference counterparts by having different start or/and end coordinates, with otherwise similar exon-intron structure. To shortlist novel transcripts with distinct intronic coordinates only, simply set the “refine.by” argument to “intron”:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">custom_new.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subsetNewTranscripts.html">subsetNewTranscripts</a></span><span class="op">(</span><span class="va">custom_matched_2.gtf</span>, <span class="va">ref.gtf</span>, refine.by <span class="op">=</span> <span class="st">"intron"</span><span class="op">)</span>  
<span class="co">#&gt; Removing transcripts with exact exon coordinates</span>
<span class="co">#&gt; Removing transcripts with exact intron coordinates</span>

<span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">custom_new.gtf</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span><span class="op">)</span>  </code></pre></div>
<p><img src="factR_files/figure-html/subsettx2-1.png" width="700"></p>
<p>We will use the custom_new.gtf object in the rest of the workflow.</p>
</div>
<div id="constructing-cds-information" class="section level2">
<h2 class="hasAnchor">
<a href="#constructing-cds-information" class="anchor"></a>Constructing CDS information</h2>
<p>Functional annotation of newly assembled transcripts in <em>factR</em> begins by deducing their protein-coding sequences (CDSs). To search for putative CDSs, <em>factR</em> requires a genome sequence file, which can be obtained using R packages or downloaded from online databases (e.g. UCSC, GENCODE or Ensembl). Three alternative ways to retrieve mouse genomic sequence are described below.</p>
<ol style="list-style-type: decimal">
<li>Using <em>BSgenome</em><br>
This package supports most sequenced genomes. Mouse mm10 sequence can be downloaded as follows:<br><!-- Installation code below is for demonstration and not evaluated -->
</li>
</ol>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>  
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>  
  
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"BSgenome.Mmusculus.UCSC.mm10"</span><span class="op">)</span>  </code></pre></div>
<p>and loaded into R environment:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Mmusculus.UCSC.mm10</span><span class="op">)</span>
<span class="va">Mmusculus</span> <span class="op">&lt;-</span> <span class="va">BSgenome.Mmusculus.UCSC.mm10</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Using <em>AnnotationHub</em>
</li>
</ol>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span>   
<span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span>  
<span class="co">#&gt; snapshotDate(): 2020-10-27</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"mm10"</span>,<span class="st">"2bit"</span><span class="op">)</span><span class="op">)</span>   
<span class="co">#&gt; AnnotationHub with 1 record</span>
<span class="co">#&gt; # snapshotDate(): 2020-10-27</span>
<span class="co">#&gt; # names(): AH14005</span>
<span class="co">#&gt; # $dataprovider: UCSC</span>
<span class="co">#&gt; # $species: Mus musculus</span>
<span class="co">#&gt; # $rdataclass: TwoBitFile</span>
<span class="co">#&gt; # $rdatadateadded: 2014-12-15</span>
<span class="co">#&gt; # $title: mm10.2bit</span>
<span class="co">#&gt; # $description: UCSC 2 bit file for mm10 </span>
<span class="co">#&gt; # $taxonomyid: 10090</span>
<span class="co">#&gt; # $genome: mm10</span>
<span class="co">#&gt; # $sourcetype: TwoBit</span>
<span class="co">#&gt; # $sourceurl: http://hgdownload.cse.ucsc.edu/goldenpath/mm10/bigZips/mm10.2bit</span>
<span class="co">#&gt; # $sourcesize: NA</span>
<span class="co">#&gt; # $tags: c("2bit", "UCSC", "genome") </span>
<span class="co">#&gt; # retrieve record with 'object[["AH14005"]]'</span></code></pre></div>
<!-- Code below not evaluated as it serves as an alternative method of retrieving genome data -->
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Retrieve mouse genome</span>
<span class="va">Mmusculus</span> <span class="op">&lt;-</span> <span class="va">ah</span><span class="op">[[</span><span class="st">'AH14005'</span><span class="op">]</span><span class="op">]</span>  </code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Downloading from a database (e.g. GENCODE)<br><!-- Code below not evaluated as it serves as an alternative method of retrieving genome data -->
</li>
</ol>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="st">"ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M25/GRCm38.primary_assembly.genome.fa.gz"</span>,  
              <span class="st">"GRCm38.primary_assembly.genome.fa.gz"</span><span class="op">)</span>  
<span class="va">Mmusculus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/importFASTA.html">importFASTA</a></span><span class="op">(</span><span class="st">"GRCm38.primary_assembly.genome.fa.gz"</span><span class="op">)</span>  </code></pre></div>
<p>Once the genome sequence object is ready, <em>factR</em> can predict CDSs using its buildCDS() function and reference transcriptome data as a guide. buildCDS() first generates a database of previously annotated ATGs and uses this information to search for a potential translation start sites in query transcripts. buildCDS() then deduces the CDS and appends its coordinates to the custom transcriptome object. Let’s run this function for our novel transcripts:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">custom_new_CDS.gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/buildCDS.html">buildCDS</a></span><span class="op">(</span><span class="va">custom_new.gtf</span>, <span class="va">ref.gtf</span>, <span class="va">Mmusculus</span><span class="op">)</span>  
<span class="co">#&gt; Searching for reference mRNAs in query</span>
<span class="co">#&gt; No reference mRNAs found</span>
<span class="co">#&gt; Building database of annotated ATG codons</span>
<span class="co">#&gt; Selecting best ATG start codon for remaining transcripts and determining open-reading frame</span>
<span class="co">#&gt; 15254 new CDSs constructed</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Summary: Out of 38479 transcripts in `custom_new.gtf`, 15254 transcript CDSs were built</span></code></pre></div>
<p>Note that the novel <em>Zfr</em> transcripts have been updated with information about likely CDSs (dark blue) and untranslated regions (light blue) :</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span><span class="op">)</span>  </code></pre></div>
<p><img src="factR_files/figure-html/viewafterCDS-1.png" width="700"></p>
<p>We can display exonic regions and CDSs more clearly (at the expense of loosing their bona fide genomic coordinates) by setting rescale_intron argument to TRUE.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span>, rescale_introns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>  </code></pre></div>
<p><img src="factR_files/figure-html/viewafterCDSscale-1.png" width="700"></p>
<p>For comparison, here is the CDS situation in the reference Zfr transcripts:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/viewTranscripts.html">viewTranscripts</a></span><span class="op">(</span><span class="va">ref.gtf</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span>, rescale_introns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>  </code></pre></div>
<p><img src="factR_files/figure-html/viewrefCDS-1.png" width="700"></p>
</div>
<div id="predicting-nmd" class="section level2">
<h2 class="hasAnchor">
<a href="#predicting-nmd" class="anchor"></a>Predicting NMD</h2>
<p>To explore possible susceptibility of newly identified mRNA isoforms to NMD, we use the <code>predictNMD</code> function:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">NMDprediction.out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictNMD.html">predictNMD</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span><span class="op">)</span>  
<span class="co">#&gt; Predicting NMD sensitivities for 15254 mRNAs</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">NMDprediction.out</span><span class="op">)</span>  
<span class="co">#&gt; # A tibble: 6 x 5</span>
<span class="co">#&gt;   transcript            stop_to_lastEJ num_of_downEJs `3'UTR_length` is_NMD</span>
<span class="co">#&gt;   &lt;chr&gt;                          &lt;dbl&gt;          &lt;int&gt;          &lt;dbl&gt; &lt;lgl&gt; </span>
<span class="co">#&gt; 1 ENSMUST00000011407.7            -289              0           1517 FALSE </span>
<span class="co">#&gt; 2 ENSMUST00000022695.15             56              1           2726 TRUE  </span>
<span class="co">#&gt; 3 ENSMUST00000023393.14             97              1            825 TRUE  </span>
<span class="co">#&gt; 4 ENSMUST00000026677.3              NA              0            496 FALSE </span>
<span class="co">#&gt; 5 ENSMUST00000032813.9            4372             34           4582 TRUE  </span>
<span class="co">#&gt; 6 ENSMUST00000033578.6              NA              0            599 FALSE</span></code></pre></div>
<p><code>predictNMD</code> outputs a data frame containing information on features that may promote NMD and predicts NMD sensitivity for each CDS-containing transcript based on the distance between the stop codon and the last exon-exon junction.</p>
<p>To identify putative NMD targets for specific genes (e.g. <em>Zfr</em>), run the following:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">NMDprediction.Zfr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictNMD.html">predictNMD</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span><span class="op">)</span> 
<span class="co">#&gt; Predicting NMD sensitivities for 7 mRNAs</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">NMDprediction.Zfr</span><span class="op">)</span>  
<span class="co">#&gt; # A tibble: 6 x 5</span>
<span class="co">#&gt;   transcript     stop_to_lastEJ num_of_downEJs `3'UTR_length` is_NMD</span>
<span class="co">#&gt;   &lt;chr&gt;                   &lt;dbl&gt;          &lt;int&gt;          &lt;dbl&gt; &lt;lgl&gt; </span>
<span class="co">#&gt; 1 MSTRG.14649.1            3857             11           8774 TRUE  </span>
<span class="co">#&gt; 2 MSTRG.14649.10           -177              0           4740 FALSE </span>
<span class="co">#&gt; 3 MSTRG.14649.11             41              1           4958 FALSE </span>
<span class="co">#&gt; 4 MSTRG.14649.12           -300              0           2242 FALSE </span>
<span class="co">#&gt; 5 MSTRG.14649.3            3857             11           8774 TRUE  </span>
<span class="co">#&gt; 6 MSTRG.14649.4            -300              0           2242 FALSE</span></code></pre></div>
</div>
<div id="predicting-protein-domains" class="section level2">
<h2 class="hasAnchor">
<a href="#predicting-protein-domains" class="anchor"></a>Predicting protein domains</h2>
<p><em>factR</em> can also inspect domain structure of protein products encoded by newly identified mRNA isoforms using its <code>predictDomains</code> tool:<br>
(Note: code below will predict domains on all proteins and will require a substantial amount of time to complete) <!-- Code below not evaluated as it takes substantial amount of time to complete --></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">domains.out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictDomains.html">predictDomains</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="va">Mmusculus</span><span class="op">)</span>  </code></pre></div>
<p>The above code returns a data frame containing the name, expected value and position of predicted protein domain families within each transcript.</p>
<p>Since predictDomains() connects to the online PFAM database, it may require a substantial amount of time and stable internet connection to query multiple transcripts simultaneously. To quickly explore this functionality, users may prefer to begin with a relatively small subset of transcripts. For example, to predict the domain structure for all transcripts of the <em>Zfr</em> gene, run:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/predictDomains.html">predictDomains</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="va">Mmusculus</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span><span class="op">)</span>  
<span class="co">#&gt; Checking CDSs and translating protein sequences</span>
<span class="co">#&gt; Predicting domain families for 7 proteins</span>
<span class="co">#&gt; # A tibble: 6 x 5</span>
<span class="co">#&gt;   transcript    description                      eval  begin   end</span>
<span class="co">#&gt;   &lt;chr&gt;         &lt;chr&gt;                            &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt;</span>
<span class="co">#&gt; 1 MSTRG.14649.1 HkH motif-containing C2H2 finger 0.044   303   364</span>
<span class="co">#&gt; 2 MSTRG.14649.1 HkH motif-containing C2H2 finger 0.065   371   414</span>
<span class="co">#&gt; 3 MSTRG.14649.3 Classic zinc finger, C2H2        0.063   303   350</span>
<span class="co">#&gt; 4 MSTRG.14649.4 Classic zinc finger, C2H2        0.063   311   358</span>
<span class="co">#&gt; 5 MSTRG.14649.9 HkH motif-containing C2H2 finger 0.044   289   350</span>
<span class="co">#&gt; 6 MSTRG.14649.9 HkH motif-containing C2H2 finger 0.065   357   400</span></code></pre></div>
<p>The domain architectures for the set of query proteins can be additionally plotted by switching the argument “plot” to TRUE:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">domains.Zfr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictDomains.html">predictDomains</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="va">Mmusculus</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfr"</span>, plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>  
<span class="co">#&gt; Checking CDSs and translating protein sequences</span>
<span class="co">#&gt; Predicting domain families for 7 proteins</span></code></pre></div>
<p><img src="factR_files/figure-html/predDomain3-1.png" width="700"></p>
</div>
<div id="processing-factr-output" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-factr-output" class="anchor"></a>Processing <em>factR</em> output</h2>
<p>Here we describe several ways to summarize information-rich outputs of <code>predictNMD</code> and <code>predictDomains</code> functions.</p>
<div id="summarising-nmd-data" class="section level3">
<h3 class="hasAnchor">
<a href="#summarising-nmd-data" class="anchor"></a>Summarising NMD data</h3>
<p>Using the above NMDprediction.out object and <em>tidyverse</em> package tools, we can estimate the numbers of NMD-sensitive and non-NMD-sensitive transcripts among newly assembled RNAs:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>  
<span class="va">NMDprediction.out</span> <span class="op">%&gt;%</span>   
  <span class="fu">group_by</span><span class="op">(</span><span class="va">is_NMD</span><span class="op">)</span> <span class="op">%&gt;%</span>   
  <span class="fu">tally</span><span class="op">(</span><span class="op">)</span>  
<span class="co">#&gt; # A tibble: 2 x 2</span>
<span class="co">#&gt;   is_NMD     n</span>
<span class="co">#&gt;   &lt;lgl&gt;  &lt;int&gt;</span>
<span class="co">#&gt; 1 FALSE   7861</span>
<span class="co">#&gt; 2 TRUE    7393</span></code></pre></div>
<p>We might be interested in calculating mean 3’UTR lengths for these two groups of transcripts:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">NMDprediction.out</span> <span class="op">%&gt;%</span>   
  <span class="fu">group_by</span><span class="op">(</span><span class="va">is_NMD</span><span class="op">)</span> <span class="op">%&gt;%</span>   
  <span class="fu">summarise</span><span class="op">(</span>mean_3UTR_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">`3'UTR_length`</span><span class="op">)</span><span class="op">)</span>    
<span class="co">#&gt; # A tibble: 2 x 2</span>
<span class="co">#&gt;   is_NMD mean_3UTR_length</span>
<span class="co">#&gt;   &lt;lgl&gt;             &lt;dbl&gt;</span>
<span class="co">#&gt; 1 FALSE             1827.</span>
<span class="co">#&gt; 2 TRUE              3352.</span></code></pre></div>
<p>To compare the prevalence of NMD regulation between newly assembled and reference transcripts, we first predict NMD susceptibility on reference transcripts:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">NMDprediction.known</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictNMD.html">predictNMD</a></span><span class="op">(</span><span class="va">ref.gtf</span><span class="op">)</span>  
<span class="co">#&gt; Predicting NMD sensitivities for 67261 mRNAs</span></code></pre></div>
<p>Next, we combine the NMDprediction.out and NMDprediction.known data frames distinguishing between new and known transcripts:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">NMDprediction.combine</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">NMDprediction.known</span>, <span class="va">NMDprediction.out</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">mutate</span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">transcript</span> <span class="op">%in%</span> <span class="va">NMDprediction.out</span><span class="op">$</span><span class="va">transcript</span>, <span class="st">"New"</span>, <span class="st">"Known"</span><span class="op">)</span><span class="op">)</span>   </code></pre></div>
<p>Lastly, the combined transcripts are categorised into 4 groups based on their source (“New” vs “Known”) and predicted NMD sensitivity, and the number of transcripts in each group is tallied:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">proportion.NMD</span> <span class="op">&lt;-</span> <span class="va">NMDprediction.combine</span> <span class="op">%&gt;%</span>   
  <span class="fu">group_by</span><span class="op">(</span><span class="va">group</span>, <span class="va">is_NMD</span><span class="op">)</span> <span class="op">%&gt;%</span>   
  <span class="fu">tally</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_transcripts"</span><span class="op">)</span>  
<span class="va">proportion.NMD</span>  
<span class="co">#&gt; # A tibble: 4 x 3</span>
<span class="co">#&gt; # Groups:   group [2]</span>
<span class="co">#&gt;   group is_NMD n_transcripts</span>
<span class="co">#&gt;   &lt;chr&gt; &lt;lgl&gt;          &lt;int&gt;</span>
<span class="co">#&gt; 1 Known FALSE          59441</span>
<span class="co">#&gt; 2 Known TRUE            7543</span>
<span class="co">#&gt; 3 New   FALSE           8132</span>
<span class="co">#&gt; 4 New   TRUE            7399</span></code></pre></div>
<p>The above data can be visualized using a stacked bar-graph:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">proportion.NMD</span> <span class="op">%&gt;%</span>   
  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">group</span>, y<span class="op">=</span><span class="va">n_transcripts</span>, fill<span class="op">=</span><span class="va">is_NMD</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>  
  <span class="fu">geom_bar</span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span>, position<span class="op">=</span><span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Paired"</span><span class="op">)</span> </code></pre></div>
<p><img src="factR_files/figure-html/NMDplot-1.png" width="700"></p>
</div>
<div id="summarising-domain-data" class="section level3">
<h3 class="hasAnchor">
<a href="#summarising-domain-data" class="anchor"></a>Summarising domain data</h3>
<p>Gain and loss of domains can alter protein function in a profound manner. To identify domains frequently added in newly predicted protein products, we first predict protein domains of known reference proteins:<br>
(Note: code below will predict domains on all proteins and will require a substantial amount of time to complete) <!-- Code below not evaluated as it takes substantial amount of time to complete --></p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">domains.known</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictDomains.html">predictDomains</a></span><span class="op">(</span><span class="va">ref.gtf</span>, <span class="va">Mmusculus</span><span class="op">)</span>  </code></pre></div>
<p>Next, we combine the domains.out and domains.known data frames distinguishing between new and known proteins:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">domains.combine</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">domains.known</span>, <span class="va">domains.out</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">mutate</span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">transcript</span> <span class="op">%in%</span> <span class="va">domains.out</span><span class="op">$</span><span class="va">transcript</span>, <span class="st">"New"</span>, <span class="st">"Known"</span><span class="op">)</span><span class="op">)</span>   </code></pre></div>
<p>To enable grouping of protein isoforms by gene names, let’s append gene information to our combined data frame:</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># generate a transcript to gene df  </span>
<span class="va">t2g</span> <span class="op">&lt;-</span> <span class="va">custom.gtf</span> <span class="op">%&gt;%</span>  
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">select</span><span class="op">(</span><span class="va">gene_id</span>, <span class="va">gene_name</span>, transcript<span class="op">=</span><span class="va">transcript_id</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span>   
  
<span class="co"># combine t2g with domains.combine   </span>
<span class="va">domains.combine</span> <span class="op">&lt;-</span> <span class="va">domains.combine</span> <span class="op">%&gt;%</span>  
  <span class="fu">left_join</span><span class="op">(</span><span class="va">t2g</span><span class="op">)</span>  

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">domains.combine</span><span class="op">)</span>
<span class="co">#&gt;              transcript                                  description     eval</span>
<span class="co">#&gt; 1  ENSMUST00000000001.4                                   G proteins 2.07e-08</span>
<span class="co">#&gt; 2  ENSMUST00000000001.4                                   G proteins 2.07e-08</span>
<span class="co">#&gt; 3  ENSMUST00000000001.4 Transducin (alpha subunit), insertion domain 7.13e-07</span>
<span class="co">#&gt; 4 ENSMUST00000000003.13                 Retinol binding protein-like 2.00e-04</span>
<span class="co">#&gt; 5  ENSMUST00000000010.8                                  Homeodomain 2.72e-05</span>
<span class="co">#&gt; 6 ENSMUST00000000033.11                                 Insulin-like 2.74e-05</span>
<span class="co">#&gt;   begin end group               gene_id gene_name</span>
<span class="co">#&gt; 1    32  61 Known           MSTRG.28079     Gnai3</span>
<span class="co">#&gt; 2   179 349 Known           MSTRG.28079     Gnai3</span>
<span class="co">#&gt; 3    61 181 Known           MSTRG.28079     Gnai3</span>
<span class="co">#&gt; 4    25 170 Known ENSMUSG00000000003.15      Pbsn</span>
<span class="co">#&gt; 5   172 245 Known  ENSMUSG00000020875.9     Hoxb9</span>
<span class="co">#&gt; 6    17  91 Known           MSTRG.40733      Igf2</span></code></pre></div>
<p>We then generate a non-redundant list of N-to-C-ordered domain architectures for each gene</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">domains.combine.architecture</span> <span class="op">&lt;-</span> <span class="va">domains.combine</span> <span class="op">%&gt;%</span>  
  <span class="fu">group_by</span><span class="op">(</span><span class="va">transcript</span><span class="op">)</span> <span class="op">%&gt;%</span>   
  <span class="fu">arrange</span><span class="op">(</span><span class="va">begin</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">mutate</span><span class="op">(</span>domain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">description</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>   
  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">description</span>, <span class="op">-</span><span class="va">begin</span>, <span class="op">-</span><span class="va">end</span>, <span class="op">-</span><span class="va">eval</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>   
  <span class="fu">arrange</span><span class="op">(</span><span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span>   
  <span class="fu">distinct</span><span class="op">(</span><span class="va">gene_id</span>, <span class="va">domain</span>,.keep_all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>   </code></pre></div>
<p>Genes with no new domain structures are filtered out:</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">domains.combine.filtered</span> <span class="op">&lt;-</span> <span class="va">domains.combine.architecture</span> <span class="op">%&gt;%</span>  
  <span class="fu">group_by</span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="st">"New"</span> <span class="op">%in%</span> <span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> </code></pre></div>
<p>Domains gained in new proteins are shortlisted:</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">domains.combine.shortlist</span> <span class="op">&lt;-</span> <span class="va">domains.combine.filtered</span> <span class="op">%&gt;%</span>  
  <span class="fu">unnest</span><span class="op">(</span><span class="va">domain</span><span class="op">)</span> <span class="op">%&gt;%</span>   
  <span class="fu">group_by</span><span class="op">(</span><span class="va">transcript</span>, <span class="va">domain</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">mutate</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>   
  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">distinct</span><span class="op">(</span><span class="va">gene_id</span>, <span class="va">group</span>, <span class="va">domain</span>, <span class="va">n</span>,.keep_all<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">group_by</span><span class="op">(</span><span class="va">gene_id</span>, <span class="va">domain</span>, <span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">n</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">group</span> <span class="op">==</span> <span class="st">"New"</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span>   </code></pre></div>
<p>Top 10 frequently gained domains can be listed as follows:</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">domains.combine.top</span> <span class="op">&lt;-</span> <span class="va">domains.combine.shortlist</span> <span class="op">%&gt;%</span>  
  <span class="fu">distinct</span><span class="op">(</span><span class="va">gene_id</span>, <span class="va">domain</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">group_by</span><span class="op">(</span><span class="va">domain</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">tally</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">slice_max</span><span class="op">(</span><span class="va">n</span>, n<span class="op">=</span><span class="fl">10</span>, with_ties <span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>   
  
<span class="va">domains.combine.top</span>   
<span class="co">#&gt; # A tibble: 14 x 2</span>
<span class="co">#&gt;    domain                                             n</span>
<span class="co">#&gt;    &lt;chr&gt;                                          &lt;int&gt;</span>
<span class="co">#&gt;  1 Classic zinc finger, C2H2                         12</span>
<span class="co">#&gt;  2 Armadillo repeat                                   6</span>
<span class="co">#&gt;  3 Ngr ectodomain-like                                5</span>
<span class="co">#&gt;  4 PBS lyase HEAT-like repeat                         5</span>
<span class="co">#&gt;  5 KRAB domain (Kruppel-associated box)               4</span>
<span class="co">#&gt;  6 Quinoprotein alcohol dehydrogenase-like            4</span>
<span class="co">#&gt;  7 C1 set domains (antibody constant domain-like)     3</span>
<span class="co">#&gt;  8 Fibronectin type III                               3</span>
<span class="co">#&gt;  9 Outer arm dynein light chain 1                     3</span>
<span class="co">#&gt; 10 Protein kinases, catalytic subunit                 3</span>
<span class="co">#&gt; 11 Rhodopsin-like                                     3</span>
<span class="co">#&gt; 12 RING finger domain, C3HC4                          3</span>
<span class="co">#&gt; 13 Spectrin repeat                                    3</span>
<span class="co">#&gt; 14 WD40-repeat                                        3</span></code></pre></div>
<p>We can summarise the above analysis as a bar chart:</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">domains.combine.top</span> <span class="op">%&gt;%</span>   
  <span class="fu">mutate</span><span class="op">(</span>domain <span class="op">=</span> <span class="fu">fct_reorder</span><span class="op">(</span><span class="va">domain</span>, <span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">domain</span>, y <span class="op">=</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>  
  <span class="fu">geom_bar</span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span>  
  <span class="fu">coord_flip</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>   
  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Number of genes"</span><span class="op">)</span>  </code></pre></div>
<p><img src="factR_files/figure-html/plotdomains-1.png" width="700"></p>
<p>We can also list the first 10 genes gaining the C2H2 domain:</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">domains.combine.shortlist</span> <span class="op">%&gt;%</span>   
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">domain</span> <span class="op">==</span> <span class="st">"Classic zinc finger, C2H2"</span><span class="op">)</span> <span class="op">%&gt;%</span>   
  <span class="fu">distinct</span><span class="op">(</span><span class="va">gene_id</span>, <span class="va">gene_name</span><span class="op">)</span>  
<span class="co">#&gt; # A tibble: 12 x 2</span>
<span class="co">#&gt;    gene_id     gene_name</span>
<span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;    </span>
<span class="co">#&gt;  1 MSTRG.18160 Gm10226  </span>
<span class="co">#&gt;  2 MSTRG.41926 &lt;NA&gt;     </span>
<span class="co">#&gt;  3 MSTRG.4703  &lt;NA&gt;     </span>
<span class="co">#&gt;  4 MSTRG.4722  Zfp873   </span>
<span class="co">#&gt;  5 MSTRG.17814 &lt;NA&gt;     </span>
<span class="co">#&gt;  6 MSTRG.18100 &lt;NA&gt;     </span>
<span class="co">#&gt;  7 MSTRG.25421 &lt;NA&gt;     </span>
<span class="co">#&gt;  8 MSTRG.26085 &lt;NA&gt;     </span>
<span class="co">#&gt;  9 MSTRG.4713  &lt;NA&gt;     </span>
<span class="co">#&gt; 10 MSTRG.38364 &lt;NA&gt;     </span>
<span class="co">#&gt; 11 MSTRG.47576 Yy2      </span>
<span class="co">#&gt; 12 MSTRG.14649 &lt;NA&gt;</span></code></pre></div>
<p>Finally, to visualize domain architectures of new and known protein isoforms produced from a gene of interest (say, <em>Zfp972</em>):</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">domains.Zfp972</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictDomains.html">predictDomains</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="va">ref.gtf</span><span class="op">)</span>, <span class="va">Mmusculus</span>, <span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Zfp972"</span>, plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>  
<span class="co">#&gt; Checking CDSs and translating protein sequences</span>
<span class="co">#&gt; Predicting domain families for 4 proteins</span></code></pre></div>
<p><img src="factR_files/figure-html/showZfp972-1.png" width="700"></p>
</div>
</div>
<div id="export-output-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#export-output-objects" class="anchor"></a>Export output objects</h2>
<p>Annotated <em>custom_new_CDS.gtf</em> object can be exported to a GTF file as follows:</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/export.html">export</a></span><span class="op">(</span><span class="va">custom_new_CDS.gtf</span>, <span class="st">"Custom_new.gtf"</span>, format <span class="op">=</span> <span class="st">"GTF"</span><span class="op">)</span>  </code></pre></div>
<p>Finally, to export <em>NMDprediction.out</em> and <em>domains.out</em> data frames as tab-delimited text files, run the following:</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">NMDprediction.out</span>, <span class="st">"Custom_new_NMD.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, quote <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>  
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">domains.out</span>, <span class="st">"Custom_new_domains.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, quote <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>  </code></pre></div>
</div>
</div>
<div id="citing-factr" class="section level1">
<h1 class="hasAnchor">
<a href="#citing-factr" class="anchor"></a>Citing <em>factR</em>
</h1>
<p>Please cite <em>factR</em> if you find it useful:<br><!-- Add citation --></p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h1>
<p>This workflow was conducted on:</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>  
<span class="co">#&gt; R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    parallel  stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] forcats_0.5.0                      stringr_1.4.0                     </span>
<span class="co">#&gt;  [3] dplyr_1.0.2                        purrr_0.3.4                       </span>
<span class="co">#&gt;  [5] readr_1.4.0                        tidyr_1.1.2                       </span>
<span class="co">#&gt;  [7] tibble_3.0.4                       ggplot2_3.3.3                     </span>
<span class="co">#&gt;  [9] tidyverse_1.3.0                    GenomicFeatures_1.42.1            </span>
<span class="co">#&gt; [11] AnnotationDbi_1.52.0               Biobase_2.50.0                    </span>
<span class="co">#&gt; [13] BSgenome.Mmusculus.UCSC.mm10_1.4.0 BSgenome_1.58.0                   </span>
<span class="co">#&gt; [15] rtracklayer_1.50.0                 GenomicRanges_1.42.0              </span>
<span class="co">#&gt; [17] GenomeInfoDb_1.26.2                Biostrings_2.58.0                 </span>
<span class="co">#&gt; [19] XVector_0.30.0                     IRanges_2.24.1                    </span>
<span class="co">#&gt; [21] S4Vectors_0.28.1                   AnnotationHub_2.22.0              </span>
<span class="co">#&gt; [23] BiocFileCache_1.14.0               dbplyr_2.0.0                      </span>
<span class="co">#&gt; [25] BiocGenerics_0.36.0                factR_0.99.0                      </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] colorspace_2.0-0              ellipsis_0.3.1               </span>
<span class="co">#&gt;   [3] rprojroot_2.0.2               fs_1.5.0                     </span>
<span class="co">#&gt;   [5] rstudioapi_0.13               farver_2.0.3                 </span>
<span class="co">#&gt;   [7] bit64_4.0.5                   interactiveDisplayBase_1.28.0</span>
<span class="co">#&gt;   [9] fansi_0.4.1                   lubridate_1.7.9.2            </span>
<span class="co">#&gt;  [11] xml2_1.3.2                    knitr_1.30                   </span>
<span class="co">#&gt;  [13] jsonlite_1.7.2                wiggleplotr_1.14.0           </span>
<span class="co">#&gt;  [15] Rsamtools_2.6.0               broom_0.7.3                  </span>
<span class="co">#&gt;  [17] shiny_1.5.0                   BiocManager_1.30.10          </span>
<span class="co">#&gt;  [19] compiler_4.0.3                httr_1.4.2                   </span>
<span class="co">#&gt;  [21] backports_1.2.1               assertthat_0.2.1             </span>
<span class="co">#&gt;  [23] Matrix_1.2-18                 fastmap_1.0.1                </span>
<span class="co">#&gt;  [25] cli_2.2.0                     later_1.1.0.1                </span>
<span class="co">#&gt;  [27] htmltools_0.5.0               prettyunits_1.1.1            </span>
<span class="co">#&gt;  [29] tools_4.0.3                   gtable_0.3.0                 </span>
<span class="co">#&gt;  [31] glue_1.4.2                    GenomeInfoDbData_1.2.4       </span>
<span class="co">#&gt;  [33] rappdirs_0.3.1                Rcpp_1.0.5                   </span>
<span class="co">#&gt;  [35] cellranger_1.1.0              pkgdown_1.6.1                </span>
<span class="co">#&gt;  [37] vctrs_0.3.6                   drawProteins_1.10.0          </span>
<span class="co">#&gt;  [39] xfun_0.20                     ps_1.5.0                     </span>
<span class="co">#&gt;  [41] rvest_0.3.6                   mime_0.9                     </span>
<span class="co">#&gt;  [43] lifecycle_0.2.0               XML_3.99-0.5                 </span>
<span class="co">#&gt;  [45] zlibbioc_1.36.0               scales_1.1.1                 </span>
<span class="co">#&gt;  [47] ragg_0.4.0                    hms_0.5.3                    </span>
<span class="co">#&gt;  [49] promises_1.1.1                MatrixGenerics_1.2.0         </span>
<span class="co">#&gt;  [51] SummarizedExperiment_1.20.0   RColorBrewer_1.1-2           </span>
<span class="co">#&gt;  [53] yaml_2.2.1                    curl_4.3                     </span>
<span class="co">#&gt;  [55] pbapply_1.4-3                 memoise_1.1.0                </span>
<span class="co">#&gt;  [57] biomaRt_2.46.0                stringi_1.5.3                </span>
<span class="co">#&gt;  [59] RSQLite_2.2.1                 BiocVersion_3.12.0           </span>
<span class="co">#&gt;  [61] desc_1.2.0                    BiocParallel_1.24.1          </span>
<span class="co">#&gt;  [63] rlang_0.4.10                  pkgconfig_2.0.3              </span>
<span class="co">#&gt;  [65] systemfonts_0.3.2             matrixStats_0.57.0           </span>
<span class="co">#&gt;  [67] bitops_1.0-6                  evaluate_0.14                </span>
<span class="co">#&gt;  [69] lattice_0.20-41               labeling_0.4.2               </span>
<span class="co">#&gt;  [71] GenomicAlignments_1.26.0      bit_4.0.4                    </span>
<span class="co">#&gt;  [73] tidyselect_1.1.0              magrittr_2.0.1               </span>
<span class="co">#&gt;  [75] bookdown_0.21                 R6_2.5.0                     </span>
<span class="co">#&gt;  [77] generics_0.1.0                DelayedArray_0.16.0          </span>
<span class="co">#&gt;  [79] DBI_1.1.0                     pillar_1.4.7                 </span>
<span class="co">#&gt;  [81] haven_2.3.1                   withr_2.3.0                  </span>
<span class="co">#&gt;  [83] RCurl_1.98-1.2                modelr_0.1.8                 </span>
<span class="co">#&gt;  [85] crayon_1.3.4                  utf8_1.1.4                   </span>
<span class="co">#&gt;  [87] rmarkdown_2.6                 progress_1.2.2               </span>
<span class="co">#&gt;  [89] grid_4.0.3                    readxl_1.3.1                 </span>
<span class="co">#&gt;  [91] data.table_1.13.6             blob_1.2.1                   </span>
<span class="co">#&gt;  [93] rmdformats_1.0.0              reprex_0.3.0                 </span>
<span class="co">#&gt;  [95] digest_0.6.27                 xtable_1.8-4                 </span>
<span class="co">#&gt;  [97] httpuv_1.5.4                  textshaping_0.2.1            </span>
<span class="co">#&gt;  [99] openssl_1.4.3                 munsell_0.5.0                </span>
<span class="co">#&gt; [101] askpass_1.1</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Kovaka2019">
<p>Kovaka, Sam, Aleksey V. Zimin, Geo M. Pertea, Roham Razaghi, Steven L. Salzberg, and Mihaela Pertea. 2019. “Transcriptome Assembly from Long-Read Rna-Seq Alignments with Stringtie2.” <em>Genome Biology</em> 20 (1): 278. <a href="https://doi.org/10.1186/s13059-019-1910-1">https://doi.org/10.1186/s13059-019-1910-1</a>.</p>
</div>
<div id="ref-Li2018">
<p>Li, Heng. 2018. “Minimap2: Pairwise Alignment for Nucleotide Sequences.” <em>Bioinformatics</em> 34 (18): 3094–3100. <a href="https://doi.org/10.1093/bioinformatics/bty191">https://doi.org/10.1093/bioinformatics/bty191</a>.</p>
</div>
<div id="ref-Pertea2016">
<p>Pertea, Mihaela, Daehwan Kim, Geo M. Pertea, Jeffrey T. Leek, and Steven L. Salzberg. 2016. “Transcript-Level Expression Analysis of Rna-Seq Experiments with Hisat, Stringtie and Ballgown.” <em>Nature Protocols</em> 11 (9): 1650–67. <a href="https://doi.org/10.1038/nprot.2016.095">https://doi.org/10.1038/nprot.2016.095</a>.</p>
</div>
<div id="ref-Sessegolo2019">
<p>Sessegolo, Camille, Corinne Cruaud, Corinne Da Silva, Audric Cologne, Marion Dubarry, Thomas Derrien, Vincent Lacroix, and Jean-Marc Aury. 2019. “Transcriptome Profiling of Mouse Samples Using Nanopore Sequencing of cDNA and Rna Molecules.” <em>Scientific Reports</em> 9 (1): 14908. <a href="https://doi.org/10.1038/s41598-019-51470-9">https://doi.org/10.1038/s41598-019-51470-9</a>.</p>
</div>
<div id="ref-Tasic2018">
<p>Tasic, Bosiljka, Zizhen Yao, Lucas T. Graybuck, Kimberly A. Smith, Thuc Nghi Nguyen, Darren Bertagnolli, Jeff Goldy, et al. 2018. “Shared and Distinct Transcriptomic Cell Types Across Neocortical Areas.” <em>Nature</em> 563 (7729): 72–78. <a href="https://doi.org/10.1038/s41586-018-0654-5">https://doi.org/10.1038/s41586-018-0654-5</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
